<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>8 Annotation of genes | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="8 Annotation of genes | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="8 Annotation of genes | _main.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="study-dataset.html"/>
<link rel="next" href="convert-gene-symbol-to-gene-id-using-biomart..html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.26/datatables.js"></script>
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Intro</a></li>
<li class="chapter" data-level="3" data-path="extracting-data-from-ncbi-gene-expression-omnibus-databse.html"><a href="extracting-data-from-ncbi-gene-expression-omnibus-databse.html"><i class="fa fa-check"></i><b>3</b> Extracting data from NCBI Gene Expression Omnibus databse</a></li>
<li class="chapter" data-level="4" data-path="download-data-to-the-current-working-environment.html"><a href="download-data-to-the-current-working-environment.html"><i class="fa fa-check"></i><b>4</b> Download data to the current working environment</a><ul>
<li class="chapter" data-level="4.1" data-path="download-data-to-the-current-working-environment.html"><a href="download-data-to-the-current-working-environment.html#summary-of-experiment"><i class="fa fa-check"></i><b>4.1</b> Summary of experiment</a></li>
<li class="chapter" data-level="4.2" data-path="download-data-to-the-current-working-environment.html"><a href="download-data-to-the-current-working-environment.html#assigning-the-expression-and-phenotype-data"><i class="fa fa-check"></i><b>4.2</b> Assigning the expression and phenotype data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="convert-to-gds-to-expression-set.html"><a href="convert-to-gds-to-expression-set.html"><i class="fa fa-check"></i><b>5</b> convert to GDS to expression set</a></li>
<li class="chapter" data-level="6" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html"><i class="fa fa-check"></i><b>6</b> Query bioloical pathway databases</a></li>
<li class="chapter" data-level="7" data-path="study-dataset.html"><a href="study-dataset.html"><i class="fa fa-check"></i><b>7</b> Study dataset</a></li>
<li class="chapter" data-level="8" data-path="annotation-of-genes.html"><a href="annotation-of-genes.html"><i class="fa fa-check"></i><b>8</b> Annotation of genes</a></li>
<li class="chapter" data-level="9" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html"><i class="fa fa-check"></i><b>9</b> Convert gene symbol to gene ID using <code>biomaRt</code>.</a><ul>
<li class="chapter" data-level="9.1" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html#connect-to-the-selected-biomart-database-using-useensembl-function."><i class="fa fa-check"></i><b>9.1</b> 1. Connect to the selected <em>BioMart</em> database using <code>useEnsembl</code> function.</a><ul>
<li class="chapter" data-level="9.1.1" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html#list-of-available-biomart-databases-hosted-by-ensembl"><i class="fa fa-check"></i><b>9.1.1</b> List of available BioMart databases hosted by Ensembl</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html#annotate-genes"><i class="fa fa-check"></i><b>9.2</b> 2. Annotate genes</a></li>
<li class="chapter" data-level="9.3" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html#alternative-methods-for-annotation"><i class="fa fa-check"></i><b>9.3</b> Alternative methods for annotation</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="curate-biological-functions.html"><a href="curate-biological-functions.html"><i class="fa fa-check"></i><b>10</b> Curate biological functions</a></li>
<li class="chapter" data-level="11" data-path="reference.html"><a href="reference.html"><i class="fa fa-check"></i><b>11</b> Reference</a></li>
<li class="chapter" data-level="12" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html"><i class="fa fa-check"></i><b>12</b> Over-representation Analysis</a></li>
<li class="chapter" data-level="13" data-path="experiment-data.html"><a href="experiment-data.html"><i class="fa fa-check"></i><b>13</b> Experiment Data</a></li>
<li class="chapter" data-level="14" data-path="gene-set-selection.html"><a href="gene-set-selection.html"><i class="fa fa-check"></i><b>14</b> Gene set selection</a><ul>
<li class="chapter" data-level="14.1" data-path="gene-set-selection.html"><a href="gene-set-selection.html#define-differentially-expressed-gene"><i class="fa fa-check"></i><b>14.1</b> Define differentially expressed gene</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="over-representation-test.html"><a href="over-representation-test.html"><i class="fa fa-check"></i><b>15</b> Over-representation test</a><ul>
<li class="chapter" data-level="15.1" data-path="over-representation-test.html"><a href="over-representation-test.html#gene-set-1"><i class="fa fa-check"></i><b>15.1</b> Gene set 1</a></li>
<li class="chapter" data-level="15.2" data-path="over-representation-test.html"><a href="over-representation-test.html#gene-set-2"><i class="fa fa-check"></i><b>15.2</b> Gene set 2</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html"><i class="fa fa-check"></i><b>16</b> Over-respensentation analysis from R package</a><ul>
<li class="chapter" data-level="16.1" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#test-set-up-and-result-interpretations"><i class="fa fa-check"></i><b>16.1</b> Test set up and Result interpretations</a></li>
<li class="chapter" data-level="16.2" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#examples"><i class="fa fa-check"></i><b>16.2</b> Examples</a><ul>
<li class="chapter" data-level="16.2.1" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#go"><i class="fa fa-check"></i><b>16.2.1</b> GO</a></li>
<li class="chapter" data-level="16.2.2" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#kegg"><i class="fa fa-check"></i><b>16.2.2</b> KEGG</a></li>
<li class="chapter" data-level="16.2.3" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#wikipathways"><i class="fa fa-check"></i><b>16.2.3</b> WikiPathways</a></li>
<li class="chapter" data-level="16.2.4" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#reactome"><i class="fa fa-check"></i><b>16.2.4</b> Reactome</a></li>
<li class="chapter" data-level="16.2.5" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#do"><i class="fa fa-check"></i><b>16.2.5</b> DO</a></li>
<li class="chapter" data-level="16.2.6" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#disgenet"><i class="fa fa-check"></i><b>16.2.6</b> DisGeNET</a></li>
<li class="chapter" data-level="16.2.7" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#mesh"><i class="fa fa-check"></i><b>16.2.7</b> MeSH</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="pros-and-cons-of-ora.html"><a href="pros-and-cons-of-ora.html"><i class="fa fa-check"></i><b>17</b> Pros and Cons of ORA</a></li>
<li class="chapter" data-level="18" data-path="reference-1.html"><a href="reference-1.html"><i class="fa fa-check"></i><b>18</b> Reference</a></li>
<li class="chapter" data-level="19" data-path="gene-set-enrichment-analysis.html"><a href="gene-set-enrichment-analysis.html"><i class="fa fa-check"></i><b>19</b> Gene set enrichment analysis</a></li>
<li class="chapter" data-level="20" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>20</b> Data</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="annotation-of-genes" class="section level1 hasAnchor">
<h1><span class="header-section-number">8</span> Annotation of genes<a href="annotation-of-genes.html#annotation-of-genes" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>The GDS3713 dataset comes with an annotation file that has information
for genes in the array:</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb23-1" title="1"><span class="kw">cat</span>(<span class="kw">paste0</span>(<span class="kw">names</span>(annot), <span class="dt">collapse =</span> <span class="st">&#39;</span><span class="ch">\t</span><span class="st">&#39;</span>))</a></code></pre></div>
<pre><code>ID  Gene.title  Gene.symbol Gene.ID UniGene.title   UniGene.symbol  UniGene.ID  Nucleotide.Title    GI  GenBank.Accession   Platform_CLONEID    Platform_ORF    Platform_SPOTID Chromosome.location Chromosome.annotation   GO.Function GO.Process  GO.Component    GO.Function.ID  GO.Process.ID   GO.Component.ID</code></pre>
<p>One can directly query a specific gene from this file. For example, we can
look up the ICOSLG information.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb25-1" title="1"><span class="kw">datatable</span>(annot[<span class="kw">which</span>(annot<span class="op">$</span>Gene.symbol <span class="op">==</span><span class="st"> &quot;ICOSLG&quot;</span>), ],</a>
<a class="sourceLine" id="cb25-2" title="2">  <span class="dt">rownames =</span> F</a>
<a class="sourceLine" id="cb25-3" title="3">)</a></code></pre></div>
<div id="htmlwidget-4999012b56b4ed1472e8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-4999012b56b4ed1472e8">{"x":{"filter":"none","vertical":false,"data":[["211197_s_at","211198_s_at","211199_s_at","213450_s_at"],["inducible T-cell costimulator ligand","inducible T-cell costimulator ligand","inducible T-cell costimulator ligand","inducible T-cell costimulator ligand"],["ICOSLG","ICOSLG","ICOSLG","ICOSLG"],["23308","23308","23308","23308"],[null,null,null,null],[null,null,null,null],[null,null,null,null],["Homo sapiens EST from clone 34465, full insert","Homo sapiens transmembrane protein B7-H2 ICOS ligand mRNA, complete cds","Homo sapiens B7-like protein (GL50) mRNA, complete cds",null],[7799138,9858866,6983943,null],["AL355690","AF289028","AF199028",null],[null,null,null,null],[null,null,null,null],[null,null,null,null],["21q22.3","21q22.3","21q22.3","21q22.3"],["Chromosome 21, NC_000021.9 (44216980..44241447, complement)","Chromosome 21, NC_000021.9 (44216980..44241447, complement)","Chromosome 21, NC_000021.9 (44216980..44241447, complement)","Chromosome 21, NC_000021.9 (44216980..44241447, complement)"],["receptor binding","receptor binding","receptor binding","receptor binding"],["B cell activation///T cell activation///T cell costimulation///adaptive immune response///defense response///hyperosmotic response///positive regulation of activated T cell proliferation///signal transduction","B cell activation///T cell activation///T cell costimulation///adaptive immune response///defense response///hyperosmotic response///positive regulation of activated T cell proliferation///signal transduction","B cell activation///T cell activation///T cell costimulation///adaptive immune response///defense response///hyperosmotic response///positive regulation of activated T cell proliferation///signal transduction","B cell activation///T cell activation///T cell costimulation///adaptive immune response///defense response///hyperosmotic response///positive regulation of activated T cell proliferation///signal transduction"],["extracellular exosome///integral component of membrane///plasma membrane","extracellular exosome///integral component of membrane///plasma membrane","extracellular exosome///integral component of membrane///plasma membrane","extracellular exosome///integral component of membrane///plasma membrane"],["GO:0005102","GO:0005102","GO:0005102","GO:0005102"],["GO:0042113///GO:0042110///GO:0031295///GO:0002250///GO:0006952///GO:0006972///GO:0042104///GO:0007165","GO:0042113///GO:0042110///GO:0031295///GO:0002250///GO:0006952///GO:0006972///GO:0042104///GO:0007165","GO:0042113///GO:0042110///GO:0031295///GO:0002250///GO:0006952///GO:0006972///GO:0042104///GO:0007165","GO:0042113///GO:0042110///GO:0031295///GO:0002250///GO:0006952///GO:0006972///GO:0042104///GO:0007165"],["GO:0070062///GO:0016021///GO:0005886","GO:0070062///GO:0016021///GO:0005886","GO:0070062///GO:0016021///GO:0005886","GO:0070062///GO:0016021///GO:0005886"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ID<\/th>\n      <th>Gene.title<\/th>\n      <th>Gene.symbol<\/th>\n      <th>Gene.ID<\/th>\n      <th>UniGene.title<\/th>\n      <th>UniGene.symbol<\/th>\n      <th>UniGene.ID<\/th>\n      <th>Nucleotide.Title<\/th>\n      <th>GI<\/th>\n      <th>GenBank.Accession<\/th>\n      <th>Platform_CLONEID<\/th>\n      <th>Platform_ORF<\/th>\n      <th>Platform_SPOTID<\/th>\n      <th>Chromosome.location<\/th>\n      <th>Chromosome.annotation<\/th>\n      <th>GO.Function<\/th>\n      <th>GO.Process<\/th>\n      <th>GO.Component<\/th>\n      <th>GO.Function.ID<\/th>\n      <th>GO.Process.ID<\/th>\n      <th>GO.Component.ID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":8}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>However, most of time, we donâ€™t have such a file that comes with the
trancriptomic data. We will then need to annotate the genes.</p>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="study-dataset.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="convert-gene-symbol-to-gene-id-using-biomart..html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
