<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>7 Data | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="7 Data | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="7 Data | _main.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gene-set-enrichment-analysis.html"/>

<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Intro</a></li>
<li class="chapter" data-level="3" data-path="extracting-data-from-ncbi-gene-expression-omnibus-databse.html"><a href="extracting-data-from-ncbi-gene-expression-omnibus-databse.html"><i class="fa fa-check"></i><b>3</b> Extracting data from NCBI Gene Expression Omnibus databse</a><ul>
<li class="chapter" data-level="3.1" data-path="extracting-data-from-ncbi-gene-expression-omnibus-databse.html"><a href="extracting-data-from-ncbi-gene-expression-omnibus-databse.html#summary-of-experiment"><i class="fa fa-check"></i><b>3.1</b> Summary of experiment</a></li>
<li class="chapter" data-level="3.2" data-path="extracting-data-from-ncbi-gene-expression-omnibus-databse.html"><a href="extracting-data-from-ncbi-gene-expression-omnibus-databse.html#assigning-the-expression-and-phenotype-data"><i class="fa fa-check"></i><b>3.2</b> Assigning the expression and phenotype data</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html"><i class="fa fa-check"></i><b>4</b> Query bioloical pathway databases</a><ul>
<li class="chapter" data-level="4.1" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#connect-to-the-selected-biomart-database-using-useensembl-function."><i class="fa fa-check"></i><b>4.1</b> 1. Connect to the selected <em>BioMart</em> database using <code>useEnsembl</code> function.</a><ul>
<li class="chapter" data-level="4.1.1" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#list-of-available-biomart-databases-hosted-by-ensembl"><i class="fa fa-check"></i><b>4.1.1</b> List of available BioMart databases hosted by Ensembl</a></li>
</ul></li>
<li class="chapter" data-level="4.2" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#annotate-genes"><i class="fa fa-check"></i><b>4.2</b> 2. Annotate genes</a></li>
<li class="chapter" data-level="4.3" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#alternative-methods-for-annotation"><i class="fa fa-check"></i><b>4.3</b> Alternative methods for annotation</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html"><i class="fa fa-check"></i><b>5</b> Over-representation Analysis</a><ul>
<li class="chapter" data-level="5.1" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#define-differentially-expressed-gene"><i class="fa fa-check"></i><b>5.1</b> Define differentially expressed gene</a></li>
<li class="chapter" data-level="5.2" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#gene-set-1"><i class="fa fa-check"></i><b>5.2</b> Gene set 1</a></li>
<li class="chapter" data-level="5.3" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#gene-set-2"><i class="fa fa-check"></i><b>5.3</b> Gene set 2</a></li>
<li class="chapter" data-level="5.4" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#test-set-up-and-result-interpretations"><i class="fa fa-check"></i><b>5.4</b> Test set up and Result interpretations</a></li>
<li class="chapter" data-level="5.5" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#examples"><i class="fa fa-check"></i><b>5.5</b> Examples</a><ul>
<li class="chapter" data-level="5.5.1" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#go"><i class="fa fa-check"></i><b>5.5.1</b> GO</a></li>
<li class="chapter" data-level="5.5.2" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#kegg"><i class="fa fa-check"></i><b>5.5.2</b> KEGG</a></li>
<li class="chapter" data-level="5.5.3" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#wikipathways"><i class="fa fa-check"></i><b>5.5.3</b> WikiPathways</a></li>
<li class="chapter" data-level="5.5.4" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#reactome"><i class="fa fa-check"></i><b>5.5.4</b> Reactome</a></li>
<li class="chapter" data-level="5.5.5" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#do"><i class="fa fa-check"></i><b>5.5.5</b> DO</a></li>
<li class="chapter" data-level="5.5.6" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#disgenet"><i class="fa fa-check"></i><b>5.5.6</b> DisGeNET</a></li>
<li class="chapter" data-level="5.5.7" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#mesh"><i class="fa fa-check"></i><b>5.5.7</b> MeSH</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="6" data-path="gene-set-enrichment-analysis.html"><a href="gene-set-enrichment-analysis.html"><i class="fa fa-check"></i><b>6</b> Gene set enrichment analysis</a></li>
<li class="chapter" data-level="7" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>7</b> Data</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="data" class="section level1 hasAnchor">
<h1><span class="header-section-number">7</span> Data<a href="data.html#data" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<details>
<p><summary>Session Info</summary></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb3-1" title="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=C.UTF-8      
##  [2] LC_NUMERIC=C          
##  [3] LC_TIME=C.UTF-8       
##  [4] LC_COLLATE=C.UTF-8    
##  [5] LC_MONETARY=C.UTF-8   
##  [6] LC_MESSAGES=C.UTF-8   
##  [7] LC_PAPER=C.UTF-8      
##  [8] LC_NAME=C             
##  [9] LC_ADDRESS=C          
## [10] LC_TELEPHONE=C        
## [11] LC_MEASUREMENT=C.UTF-8
## [12] LC_IDENTIFICATION=C   
## 
## attached base packages:
## [1] parallel  stats4    stats    
## [4] graphics  grDevices utils    
## [7] datasets  methods   base     
## 
## other attached packages:
##  [1] MeSH.Hsa.eg.db_1.15.1
##  [2] MeSHDbi_1.28.0       
##  [3] GEOquery_2.60.0      
##  [4] dplyr_1.0.10         
##  [5] limma_3.48.2         
##  [6] DT_0.26              
##  [7] biomaRt_2.48.2       
##  [8] org.Hs.eg.db_3.13.0  
##  [9] AnnotationDbi_1.54.1 
## [10] IRanges_2.26.0       
## [11] S4Vectors_0.30.0     
## [12] Biobase_2.52.0       
## [13] BiocGenerics_0.38.0  
## [14] DOSE_3.18.3          
## [15] enrichplot_1.12.3    
## [16] clusterProfiler_4.0.5
## [17] here_1.0.1           
## 
## loaded via a namespace (and not attached):
##   [1] backports_1.2.1       
##   [2] shadowtext_0.1.1      
##   [3] fastmatch_1.1-3       
##   [4] servr_0.22            
##   [5] BiocFileCache_2.0.0   
##   [6] plyr_1.8.6            
##   [7] igraph_1.2.6          
##   [8] lazyeval_0.2.2        
##   [9] splines_4.1.0         
##  [10] BiocParallel_1.26.1   
##  [11] crosstalk_1.1.1       
##  [12] GenomeInfoDb_1.28.1   
##  [13] ggplot2_3.4.0         
##  [14] meshes_1.18.1         
##  [15] digest_0.6.27         
##  [16] yulab.utils_0.0.4     
##  [17] htmltools_0.5.5       
##  [18] GOSemSim_2.18.1       
##  [19] viridis_0.6.1         
##  [20] GO.db_3.13.0          
##  [21] fansi_0.5.0           
##  [22] checkmate_2.0.0       
##  [23] magrittr_2.0.3        
##  [24] memoise_2.0.1         
##  [25] scriptResume_0.1.4    
##  [26] tzdb_0.1.2            
##  [27] readr_2.0.0           
##  [28] Biostrings_2.60.1     
##  [29] graphlayouts_0.8.0    
##  [30] vroom_1.5.3           
##  [31] prettyunits_1.1.1     
##  [32] colorspace_2.0-2      
##  [33] blob_1.2.2            
##  [34] rappdirs_0.3.3        
##  [35] ggrepel_0.9.1         
##  [36] xfun_0.39             
##  [37] callr_3.7.3           
##  [38] crayon_1.4.1          
##  [39] RCurl_1.98-1.3        
##  [40] jsonlite_1.8.4        
##  [41] graph_1.70.0          
##  [42] scatterpie_0.1.7      
##  [43] ape_5.6-2             
##  [44] glue_1.6.2            
##  [45] polyclip_1.10-0       
##  [46] gtable_0.3.0          
##  [47] zlibbioc_1.38.0       
##  [48] XVector_0.32.0        
##  [49] webshot_0.5.2         
##  [50] graphite_1.38.0       
##  [51] scales_1.2.1          
##  [52] DBI_1.1.1             
##  [53] Rcpp_1.0.7            
##  [54] viridisLite_0.4.0     
##  [55] progress_1.2.2        
##  [56] gridGraphics_0.5-1    
##  [57] tidytree_0.3.9        
##  [58] reactome.db_1.76.0    
##  [59] bit_4.0.4             
##  [60] htmlwidgets_1.5.3     
##  [61] httr_1.4.2            
##  [62] fgsea_1.18.0          
##  [63] RColorBrewer_1.1-2    
##  [64] ellipsis_0.3.2        
##  [65] pkgconfig_2.0.3       
##  [66] XML_3.99-0.6          
##  [67] farver_2.1.0          
##  [68] sass_0.4.0            
##  [69] dbplyr_2.1.1          
##  [70] utf8_1.2.2            
##  [71] later_1.2.0           
##  [72] ggplotify_0.1.0       
##  [73] tidyselect_1.1.1      
##  [74] rlang_1.1.1           
##  [75] reshape2_1.4.4        
##  [76] munsell_0.5.0         
##  [77] tools_4.1.0           
##  [78] cachem_1.0.5          
##  [79] downloader_0.4        
##  [80] cli_3.4.1             
##  [81] generics_0.1.0        
##  [82] RSQLite_2.2.7         
##  [83] evaluate_0.15         
##  [84] stringr_1.5.0         
##  [85] fastmap_1.1.0         
##  [86] yaml_2.2.1            
##  [87] ggtree_3.0.4          
##  [88] processx_3.8.1        
##  [89] knitr_1.39            
##  [90] bit64_4.0.5           
##  [91] tidygraph_1.2.0       
##  [92] purrr_0.3.4           
##  [93] KEGGREST_1.32.0       
##  [94] ggraph_2.0.5          
##  [95] ReactomePA_1.36.0     
##  [96] nlme_3.1-152          
##  [97] mime_0.11             
##  [98] aplot_0.1.2           
##  [99] DO.db_2.9             
## [100] xml2_1.3.2            
## [101] compiler_4.1.0        
## [102] rstudioapi_0.13       
## [103] filelock_1.0.2        
## [104] curl_4.3.2            
## [105] png_0.1-7             
## [106] treeio_1.16.2         
## [107] statmod_1.4.36        
## [108] tibble_3.1.6          
## [109] tweenr_1.0.2          
## [110] bslib_0.4.2           
## [111] stringi_1.7.8         
## [112] ps_1.6.0              
## [113] highr_0.9             
## [114] lattice_0.20-44       
## [115] Matrix_1.5-3          
## [116] markdown_1.1          
## [117] vctrs_0.5.1           
## [118] pillar_1.6.2          
## [119] lifecycle_1.0.3       
## [120] jquerylib_0.1.4       
## [121] data.table_1.14.0     
## [122] cowplot_1.1.1         
## [123] bitops_1.0-7          
## [124] httpuv_1.6.1          
## [125] patchwork_1.1.1       
## [126] qvalue_2.24.0         
## [127] R6_2.5.1              
## [128] promises_1.2.0.1      
## [129] bookdown_0.34         
## [130] gridExtra_2.3         
## [131] codetools_0.2-18      
## [132] MASS_7.3-54           
## [133] assertthat_0.2.1      
## [134] rprojroot_2.0.2       
## [135] withr_2.5.0           
## [136] GenomeInfoDbData_1.2.6
## [137] hms_1.1.0             
## [138] grid_4.1.0            
## [139] ggfun_0.0.5           
## [140] MeSH.db_1.15.1        
## [141] tidyr_1.2.1           
## [142] rmarkdown_2.18        
## [143] ggforce_0.3.3</code></pre>
</details>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="gene-set-enrichment-analysis.html" class="navigation navigation-prev navigation-unique" aria-label="Previous page"><i class="fa fa-angle-left"></i></a>

    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
