<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>3 Differential gene expression analysis (DGEA) | Pathway analysis methods</title>
  <meta name="description" content="This notebook is dedicated to genetic pathway analysis methods." />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="3 Differential gene expression analysis (DGEA) | Pathway analysis methods" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This notebook is dedicated to genetic pathway analysis methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="3 Differential gene expression analysis (DGEA) | Pathway analysis methods" />
  
  <meta name="twitter:description" content="This notebook is dedicated to genetic pathway analysis methods." />
  

<meta name="author" content="Siying Huang (E0482362)" />


<meta name="date" content="2024-01-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="study-dataset.html"/>
<link rel="next" href="query-bioloical-pathway-databases.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.26/datatables.js"></script>
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Prefeace</a></li>
<li class="chapter" data-level="1" data-path="gene-sets-and-pathways.html"><a href="gene-sets-and-pathways.html"><i class="fa fa-check"></i><b>1</b> Gene sets and pathways</a>
<ul>
<li class="chapter" data-level="1.1" data-path="gene-sets-and-pathways.html"><a href="gene-sets-and-pathways.html#organization-of-this-e-book"><i class="fa fa-check"></i><b>1.1</b> Organization of this e-book</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="study-dataset.html"><a href="study-dataset.html"><i class="fa fa-check"></i><b>2</b> Study dataset</a>
<ul>
<li class="chapter" data-level="2.1" data-path="study-dataset.html"><a href="study-dataset.html#extracting-data-from-ncbi-gene-expression-omnibus-databse"><i class="fa fa-check"></i><b>2.1</b> Extracting data from NCBI Gene Expression Omnibus databse</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="study-dataset.html"><a href="study-dataset.html#summary-of-the-study"><i class="fa fa-check"></i><b>2.1.1</b> Summary of the study</a></li>
<li class="chapter" data-level="2.1.2" data-path="study-dataset.html"><a href="study-dataset.html#assigning-the-expression-and-phenotype-data"><i class="fa fa-check"></i><b>2.1.2</b> Assigning the expression and phenotype data</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="study-dataset.html"><a href="study-dataset.html#convert-to-gds-to-expression-set"><i class="fa fa-check"></i><b>2.2</b> convert to GDS to expression set</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-gene-expression-analysis-dgea.html"><a href="differential-gene-expression-analysis-dgea.html"><i class="fa fa-check"></i><b>3</b> Differential gene expression analysis (DGEA)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-gene-expression-analysis-dgea.html"><a href="differential-gene-expression-analysis-dgea.html#comparison-between-smokers-and-non-smokers"><i class="fa fa-check"></i><b>3.1</b> Comparison between smokers and non-smokers</a></li>
<li class="chapter" data-level="3.2" data-path="differential-gene-expression-analysis-dgea.html"><a href="differential-gene-expression-analysis-dgea.html#results-visualization"><i class="fa fa-check"></i><b>3.2</b> Results visualization</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html"><i class="fa fa-check"></i><b>4</b> Query bioloical pathway databases</a>
<ul>
<li class="chapter" data-level="4.1" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#gene-annotation"><i class="fa fa-check"></i><b>4.1</b> Gene annotation</a></li>
<li class="chapter" data-level="4.2" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#convert-gene-symbol-to-gene-id-using-biomart."><i class="fa fa-check"></i><b>4.2</b> Convert gene symbol to gene ID using <code>biomaRt</code>.</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#connect-to-the-selected-biomart-database-using-useensembl-function."><i class="fa fa-check"></i><b>4.2.1</b> Connect to the selected <em>BioMart</em> database using <code>useEnsembl</code> function.</a></li>
<li class="chapter" data-level="4.2.2" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#annotate-genes"><i class="fa fa-check"></i><b>4.2.2</b> Annotate genes</a></li>
<li class="chapter" data-level="4.2.3" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#alternative-methods-for-annotation"><i class="fa fa-check"></i><b>4.2.3</b> Alternative methods for annotation</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#curate-biological-functions"><i class="fa fa-check"></i><b>4.3</b> Curate biological functions</a></li>
<li class="chapter" data-level="4.4" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#reference"><i class="fa fa-check"></i><b>4.4</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html"><i class="fa fa-check"></i><b>5</b> Over-representation Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#data-prepartion"><i class="fa fa-check"></i><b>5.1</b> Data prepartion</a></li>
<li class="chapter" data-level="5.2" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#gene-set-selection"><i class="fa fa-check"></i><b>5.2</b> Gene set selection</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#define-differentially-expressed-gene"><i class="fa fa-check"></i><b>5.2.1</b> Define differentially expressed gene</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#over-representation-analysis-1"><i class="fa fa-check"></i><b>5.3</b> Over-representation analysis</a></li>
<li class="chapter" data-level="5.4" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#over-respensentation-analysis-from-r-package"><i class="fa fa-check"></i><b>5.4</b> Over-respensentation analysis from R package</a></li>
<li class="chapter" data-level="5.5" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#test-set-up-and-result-interpretations"><i class="fa fa-check"></i><b>5.5</b> Test set up and result interpretations</a></li>
<li class="chapter" data-level="5.6" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#examples"><i class="fa fa-check"></i><b>5.6</b> Examples</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#go"><i class="fa fa-check"></i><b>5.6.1</b> GO</a></li>
<li class="chapter" data-level="5.6.2" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#kegg"><i class="fa fa-check"></i><b>5.6.2</b> KEGG</a></li>
<li class="chapter" data-level="5.6.3" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#wikipathways"><i class="fa fa-check"></i><b>5.6.3</b> WikiPathways</a></li>
<li class="chapter" data-level="5.6.4" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#reactome"><i class="fa fa-check"></i><b>5.6.4</b> Reactome</a></li>
<li class="chapter" data-level="5.6.5" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#do"><i class="fa fa-check"></i><b>5.6.5</b> DO</a></li>
<li class="chapter" data-level="5.6.6" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#disgenet"><i class="fa fa-check"></i><b>5.6.6</b> DisGeNET</a></li>
<li class="chapter" data-level="5.6.7" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#mesh"><i class="fa fa-check"></i><b>5.6.7</b> MeSH</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#pros-and-cons-of-ora"><i class="fa fa-check"></i><b>5.7</b> Pros and Cons of ORA</a></li>
<li class="chapter" data-level="5.8" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#reference-1"><i class="fa fa-check"></i><b>5.8</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i>Appendix</a>
<ul>
<li class="chapter" data-level="5.9" data-path="appendix.html"><a href="appendix.html#biological-process-data-bases"><i class="fa fa-check"></i><b>5.9</b> Biological process data bases</a></li>
<li class="chapter" data-level="5.10" data-path="appendix.html"><a href="appendix.html#summary-of-currently-available-pathway-analysis-methods"><i class="fa fa-check"></i><b>5.10</b> Summary of currently available pathway analysis methods</a>
<ul>
<li class="chapter" data-level="5.10.1" data-path="appendix.html"><a href="appendix.html#over-representation-analysis-2"><i class="fa fa-check"></i><b>5.10.1</b> Over-representation analysis</a></li>
<li class="chapter" data-level="5.10.2" data-path="appendix.html"><a href="appendix.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>5.10.2</b> Gene set enrichment analysis</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pathway analysis methods</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="differential-gene-expression-analysis-dgea" class="section level1 hasAnchor" number="3">
<h1><span class="header-section-number">3</span> Differential gene expression analysis (DGEA)<a href="differential-gene-expression-analysis-dgea.html#differential-gene-expression-analysis-dgea" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<blockquote>
<p>Comparison of differential gene expression patterns in these conditions has enabled the identification of common elements that are significantly enriched in gene classes with particular functions such as protein synthesis, hormone delivery, and morphological plasticity.
- <a href="https://www.sciencedirect.com/topics/biochemistry-genetics-and-molecular-biology/differential-gene-expression">Hormones, Brain and Behavior (Third Edition), 2017</a></p>
</blockquote>
<p>Differential expression analysis result is a prerequisite for some of the
pathway analysis methods. This chapter does not aim to demonstrate how to
properly peform DGEA. Details and explainations of DEGA are omitted.</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="differential-gene-expression-analysis-dgea.html#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb14-2"><a href="differential-gene-expression-analysis-dgea.html#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(limma)</span>
<span id="cb14-3"><a href="differential-gene-expression-analysis-dgea.html#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb14-4"><a href="differential-gene-expression-analysis-dgea.html#cb14-4" aria-hidden="true" tabindex="-1"></a><span class="fu">source</span>(<span class="fu">file.path</span>(<span class="fu">here</span>(), <span class="st">&quot;src/utils.R&quot;</span>))</span>
<span id="cb14-5"><a href="differential-gene-expression-analysis-dgea.html#cb14-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-6"><a href="differential-gene-expression-analysis-dgea.html#cb14-6" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(<span class="fu">here</span>(), <span class="st">&quot;data/expr.rds&quot;</span>))</span>
<span id="cb14-7"><a href="differential-gene-expression-analysis-dgea.html#cb14-7" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">file.path</span>(<span class="fu">here</span>(), <span class="st">&quot;data/pheno.rds&quot;</span>))</span>
<span id="cb14-8"><a href="differential-gene-expression-analysis-dgea.html#cb14-8" aria-hidden="true" tabindex="-1"></a><span class="co"># data management</span></span>
<span id="cb14-9"><a href="differential-gene-expression-analysis-dgea.html#cb14-9" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> pheno <span class="sc">%&gt;%</span></span>
<span id="cb14-10"><a href="differential-gene-expression-analysis-dgea.html#cb14-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">status =</span> <span class="fu">as.factor</span>(<span class="fu">as.integer</span>(stress)))</span></code></pre></div>
<div id="comparison-between-smokers-and-non-smokers" class="section level2 hasAnchor" number="3.1">
<h2><span class="header-section-number">3.1</span> Comparison between smokers and non-smokers<a href="differential-gene-expression-analysis-dgea.html#comparison-between-smokers-and-non-smokers" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>A linear model is used with the constrast made between smokers and controls.
In this example, I use <code>limma</code> package and derived test statistics with
emperical bayes mderation on the stand errors.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="differential-gene-expression-analysis-dgea.html#cb15-1" aria-hidden="true" tabindex="-1"></a>y <span class="ot">&lt;-</span> <span class="fu">as.matrix</span>(expr[, <span class="dv">3</span><span class="sc">:</span><span class="fu">ncol</span>(expr)])</span>
<span id="cb15-2"><a href="differential-gene-expression-analysis-dgea.html#cb15-2" aria-hidden="true" tabindex="-1"></a><span class="fu">rownames</span>(y) <span class="ot">&lt;-</span> expr<span class="sc">$</span>IDENTIFIER</span>
<span id="cb15-3"><a href="differential-gene-expression-analysis-dgea.html#cb15-3" aria-hidden="true" tabindex="-1"></a>group <span class="ot">&lt;-</span> pheno[[<span class="st">&quot;status&quot;</span>]]</span>
<span id="cb15-4"><a href="differential-gene-expression-analysis-dgea.html#cb15-4" aria-hidden="true" tabindex="-1"></a>mod <span class="ot">&lt;-</span> <span class="fu">model.matrix</span>(<span class="sc">~</span> <span class="dv">0</span> <span class="sc">+</span> group)</span>
<span id="cb15-5"><a href="differential-gene-expression-analysis-dgea.html#cb15-5" aria-hidden="true" tabindex="-1"></a>fit <span class="ot">&lt;-</span> <span class="fu">lmFit</span>(y, mod)</span>
<span id="cb15-6"><a href="differential-gene-expression-analysis-dgea.html#cb15-6" aria-hidden="true" tabindex="-1"></a>contr <span class="ot">&lt;-</span> <span class="fu">makeContrasts</span>(</span>
<span id="cb15-7"><a href="differential-gene-expression-analysis-dgea.html#cb15-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">contrasts =</span> <span class="st">&quot;group1-group2&quot;</span>,</span>
<span id="cb15-8"><a href="differential-gene-expression-analysis-dgea.html#cb15-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">levels =</span> <span class="fu">colnames</span>(<span class="fu">coef</span>(fit))</span>
<span id="cb15-9"><a href="differential-gene-expression-analysis-dgea.html#cb15-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb15-10"><a href="differential-gene-expression-analysis-dgea.html#cb15-10" aria-hidden="true" tabindex="-1"></a>tmp <span class="ot">&lt;-</span> <span class="fu">contrasts.fit</span>(fit, contr)</span>
<span id="cb15-11"><a href="differential-gene-expression-analysis-dgea.html#cb15-11" aria-hidden="true" tabindex="-1"></a>ebfit <span class="ot">&lt;-</span> <span class="fu">eBayes</span>(tmp, <span class="at">robust =</span> T)</span>
<span id="cb15-12"><a href="differential-gene-expression-analysis-dgea.html#cb15-12" aria-hidden="true" tabindex="-1"></a>rst_eb <span class="ot">&lt;-</span> <span class="fu">topTable</span>(ebfit, <span class="at">number =</span> <span class="cn">Inf</span>)</span>
<span id="cb15-13"><a href="differential-gene-expression-analysis-dgea.html#cb15-13" aria-hidden="true" tabindex="-1"></a>rst <span class="ot">&lt;-</span> <span class="fu">setNames</span>(rst_eb, <span class="fu">c</span>(<span class="st">&quot;ID&quot;</span>, <span class="st">&quot;estimate&quot;</span>, <span class="st">&quot;AveExp&quot;</span>, <span class="st">&quot;t&quot;</span>, <span class="st">&quot;p.value&quot;</span>, <span class="st">&quot;FDR&quot;</span>, <span class="st">&quot;B&quot;</span>))</span></code></pre></div>
</div>
<div id="results-visualization" class="section level2 hasAnchor" number="3.2">
<h2><span class="header-section-number">3.2</span> Results visualization<a href="differential-gene-expression-analysis-dgea.html#results-visualization" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The marked genes pass the FDR significant threshold at adjusted p-value &lt;= 0.01.
This DEA result will be used for the following pathway analysis.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="differential-gene-expression-analysis-dgea.html#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="fu">par</span>(<span class="at">mfrow =</span> <span class="fu">c</span>(<span class="dv">1</span>, <span class="dv">3</span>))</span>
<span id="cb16-2"><a href="differential-gene-expression-analysis-dgea.html#cb16-2" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(rst_eb<span class="sc">$</span>logFC, <span class="at">breaks =</span> <span class="dv">50</span>, <span class="at">main =</span> <span class="st">&#39;log (Fold change)&#39;</span>)</span>
<span id="cb16-3"><a href="differential-gene-expression-analysis-dgea.html#cb16-3" aria-hidden="true" tabindex="-1"></a><span class="fu">hist</span>(rst_eb<span class="sc">$</span>P.Value, <span class="at">breaks =</span> <span class="dv">50</span>, <span class="at">main =</span> <span class="st">&#39;P values&#39;</span>)</span>
<span id="cb16-4"><a href="differential-gene-expression-analysis-dgea.html#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="fu">qqplot</span>(</span>
<span id="cb16-5"><a href="differential-gene-expression-analysis-dgea.html#cb16-5" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span><span class="fu">log10</span>(<span class="fu">runif</span>(<span class="fu">nrow</span>(rst_eb))),</span>
<span id="cb16-6"><a href="differential-gene-expression-analysis-dgea.html#cb16-6" aria-hidden="true" tabindex="-1"></a>  <span class="sc">-</span><span class="fu">log10</span>(rst_eb<span class="sc">$</span>P.Value),</span>
<span id="cb16-7"><a href="differential-gene-expression-analysis-dgea.html#cb16-7" aria-hidden="true" tabindex="-1"></a>  <span class="at">xlab =</span> <span class="st">&#39;Theoretical uniform distribution&#39;</span>,</span>
<span id="cb16-8"><a href="differential-gene-expression-analysis-dgea.html#cb16-8" aria-hidden="true" tabindex="-1"></a>  <span class="at">ylab =</span> <span class="st">&#39;-log10(P values)&#39;</span></span>
<span id="cb16-9"><a href="differential-gene-expression-analysis-dgea.html#cb16-9" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb16-10"><a href="differential-gene-expression-analysis-dgea.html#cb16-10" aria-hidden="true" tabindex="-1"></a><span class="fu">qqline</span>(<span class="sc">-</span><span class="fu">log10</span>(rst_eb<span class="sc">$</span>P.Value), <span class="at">distribution =</span> qunif, <span class="at">col =</span> <span class="st">&quot;red&quot;</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-8-1.png" width="864" /></p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="differential-gene-expression-analysis-dgea.html#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="fu">volc_plot</span>(rst, <span class="at">lab_col =</span> <span class="st">&quot;ID&quot;</span>, <span class="at">lab_method =</span> <span class="st">&quot;fdr&quot;</span>, <span class="at">fdr_thresh =</span> <span class="fl">0.01</span>)</span></code></pre></div>
<p><img src="_main_files/figure-html/unnamed-chunk-9-1.png" width="864" /></p>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="study-dataset.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="query-bioloical-pathway-databases.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
