<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Download data to the current working environment | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Download data to the current working environment | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Download data to the current working environment | _main.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="extracting-data-from-ncbi-gene-expression-omnibus-databse.html"/>
<link rel="next" href="convert-to-gds-to-expression-set.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.26/datatables.js"></script>
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Intro</a></li>
<li class="chapter" data-level="3" data-path="extracting-data-from-ncbi-gene-expression-omnibus-databse.html"><a href="extracting-data-from-ncbi-gene-expression-omnibus-databse.html"><i class="fa fa-check"></i><b>3</b> Extracting data from NCBI Gene Expression Omnibus databse</a></li>
<li class="chapter" data-level="4" data-path="download-data-to-the-current-working-environment.html"><a href="download-data-to-the-current-working-environment.html"><i class="fa fa-check"></i><b>4</b> Download data to the current working environment</a><ul>
<li class="chapter" data-level="4.1" data-path="download-data-to-the-current-working-environment.html"><a href="download-data-to-the-current-working-environment.html#summary-of-experiment"><i class="fa fa-check"></i><b>4.1</b> Summary of experiment</a></li>
<li class="chapter" data-level="4.2" data-path="download-data-to-the-current-working-environment.html"><a href="download-data-to-the-current-working-environment.html#assigning-the-expression-and-phenotype-data"><i class="fa fa-check"></i><b>4.2</b> Assigning the expression and phenotype data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="convert-to-gds-to-expression-set.html"><a href="convert-to-gds-to-expression-set.html"><i class="fa fa-check"></i><b>5</b> convert to GDS to expression set</a></li>
<li class="chapter" data-level="6" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html"><i class="fa fa-check"></i><b>6</b> Query bioloical pathway databases</a></li>
<li class="chapter" data-level="7" data-path="study-dataset.html"><a href="study-dataset.html"><i class="fa fa-check"></i><b>7</b> Study dataset</a></li>
<li class="chapter" data-level="8" data-path="annotation-of-genes.html"><a href="annotation-of-genes.html"><i class="fa fa-check"></i><b>8</b> Annotation of genes</a></li>
<li class="chapter" data-level="9" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html"><i class="fa fa-check"></i><b>9</b> Convert gene symbol to gene ID using <code>biomaRt</code>.</a><ul>
<li class="chapter" data-level="9.1" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html#connect-to-the-selected-biomart-database-using-useensembl-function."><i class="fa fa-check"></i><b>9.1</b> 1. Connect to the selected <em>BioMart</em> database using <code>useEnsembl</code> function.</a><ul>
<li class="chapter" data-level="9.1.1" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html#list-of-available-biomart-databases-hosted-by-ensembl"><i class="fa fa-check"></i><b>9.1.1</b> List of available BioMart databases hosted by Ensembl</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html#annotate-genes"><i class="fa fa-check"></i><b>9.2</b> 2. Annotate genes</a></li>
<li class="chapter" data-level="9.3" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html#alternative-methods-for-annotation"><i class="fa fa-check"></i><b>9.3</b> Alternative methods for annotation</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="curate-biological-functions.html"><a href="curate-biological-functions.html"><i class="fa fa-check"></i><b>10</b> Curate biological functions</a></li>
<li class="chapter" data-level="11" data-path="reference.html"><a href="reference.html"><i class="fa fa-check"></i><b>11</b> Reference</a></li>
<li class="chapter" data-level="12" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html"><i class="fa fa-check"></i><b>12</b> Over-representation Analysis</a></li>
<li class="chapter" data-level="13" data-path="experiment-data.html"><a href="experiment-data.html"><i class="fa fa-check"></i><b>13</b> Experiment Data</a></li>
<li class="chapter" data-level="14" data-path="gene-set-selection.html"><a href="gene-set-selection.html"><i class="fa fa-check"></i><b>14</b> Gene set selection</a><ul>
<li class="chapter" data-level="14.1" data-path="gene-set-selection.html"><a href="gene-set-selection.html#define-differentially-expressed-gene"><i class="fa fa-check"></i><b>14.1</b> Define differentially expressed gene</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="over-representation-test.html"><a href="over-representation-test.html"><i class="fa fa-check"></i><b>15</b> Over-representation test</a><ul>
<li class="chapter" data-level="15.1" data-path="over-representation-test.html"><a href="over-representation-test.html#gene-set-1"><i class="fa fa-check"></i><b>15.1</b> Gene set 1</a></li>
<li class="chapter" data-level="15.2" data-path="over-representation-test.html"><a href="over-representation-test.html#gene-set-2"><i class="fa fa-check"></i><b>15.2</b> Gene set 2</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html"><i class="fa fa-check"></i><b>16</b> Over-respensentation analysis from R package</a><ul>
<li class="chapter" data-level="16.1" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#test-set-up-and-result-interpretations"><i class="fa fa-check"></i><b>16.1</b> Test set up and Result interpretations</a></li>
<li class="chapter" data-level="16.2" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#examples"><i class="fa fa-check"></i><b>16.2</b> Examples</a><ul>
<li class="chapter" data-level="16.2.1" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#go"><i class="fa fa-check"></i><b>16.2.1</b> GO</a></li>
<li class="chapter" data-level="16.2.2" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#kegg"><i class="fa fa-check"></i><b>16.2.2</b> KEGG</a></li>
<li class="chapter" data-level="16.2.3" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#wikipathways"><i class="fa fa-check"></i><b>16.2.3</b> WikiPathways</a></li>
<li class="chapter" data-level="16.2.4" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#reactome"><i class="fa fa-check"></i><b>16.2.4</b> Reactome</a></li>
<li class="chapter" data-level="16.2.5" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#do"><i class="fa fa-check"></i><b>16.2.5</b> DO</a></li>
<li class="chapter" data-level="16.2.6" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#disgenet"><i class="fa fa-check"></i><b>16.2.6</b> DisGeNET</a></li>
<li class="chapter" data-level="16.2.7" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#mesh"><i class="fa fa-check"></i><b>16.2.7</b> MeSH</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="pros-and-cons-of-ora.html"><a href="pros-and-cons-of-ora.html"><i class="fa fa-check"></i><b>17</b> Pros and Cons of ORA</a></li>
<li class="chapter" data-level="18" data-path="reference-1.html"><a href="reference-1.html"><i class="fa fa-check"></i><b>18</b> Reference</a></li>
<li class="chapter" data-level="19" data-path="gene-set-enrichment-analysis.html"><a href="gene-set-enrichment-analysis.html"><i class="fa fa-check"></i><b>19</b> Gene set enrichment analysis</a></li>
<li class="chapter" data-level="20" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>20</b> Data</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="download-data-to-the-current-working-environment" class="section level1 hasAnchor">
<h1><span class="header-section-number">4</span> Download data to the current working environment<a href="download-data-to-the-current-working-environment.html#download-data-to-the-current-working-environment" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>Using <code>GEOquery</code> library to retrieve data from NCBI database. The GDS dataset
is stored in the <code>gds</code> R4 class object.</p>
<div class="sourceCode" id="cb4"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb4-1" title="1">gds &lt;-<span class="st"> </span><span class="kw">getGEO</span>(<span class="st">&quot;GDS3713&quot;</span>)</a>
<a class="sourceLine" id="cb4-2" title="2"><span class="co"># gds@header # meta data of the experiment</span></a>
<a class="sourceLine" id="cb4-3" title="3"><span class="co"># gds@dataTable@table # expression data</span></a>
<a class="sourceLine" id="cb4-4" title="4"><span class="co"># gds@dataTable@columns # phenotype data</span></a></code></pre></div>
<p>structure of the gds object</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb5-1" title="1"><span class="kw">str</span>(gds, <span class="dt">max.level =</span> <span class="dv">3</span>)</a></code></pre></div>
<pre><code>Formal class &#39;GDS&#39; [package &quot;GEOquery&quot;] with 3 slots
  ..@ gpl      :Formal class &#39;GPL&#39; [package &quot;GEOquery&quot;] with 2 slots
  ..@ dataTable:Formal class &#39;GEODataTable&#39; [package &quot;GEOquery&quot;] with 2 slots
  ..@ header   :List of 23
  .. ..$ channel_count           : chr &quot;1&quot;
  .. ..$ dataset_id              : chr [1:2] &quot;GDS3713&quot; &quot;GDS3713&quot;
  .. ..$ description             : chr [1:3] &quot;Analysis of peripheral circulating B cells from smoking and non-smoking healthy US white females. B cells are d&quot;| __truncated__ &quot;control&quot; &quot;cigarette smoke&quot;
  .. ..$ email                   : chr &quot;geo@ncbi.nlm.nih.gov&quot;
  .. ..$ feature_count           : chr &quot;22283&quot;
  .. ..$ institute               : chr &quot;NCBI NLM NIH&quot;
  .. ..$ name                    : chr &quot;Gene Expression Omnibus (GEO)&quot;
  .. ..$ order                   : chr &quot;none&quot;
  .. ..$ platform                : chr &quot;GPL96&quot;
  .. ..$ platform_organism       : chr &quot;Homo sapiens&quot;
  .. ..$ platform_technology_type: chr &quot;in situ oligonucleotide&quot;
  .. ..$ pubmed_id               : chr &quot;20217071&quot;
  .. ..$ ref                     : chr &quot;Nucleic Acids Res. 2005 Jan 1;33 Database Issue:D562-6&quot;
  .. ..$ reference_series        : chr &quot;GSE18723&quot;
  .. ..$ sample_count            : chr &quot;79&quot;
  .. ..$ sample_id               : chr [1:2] &quot;GSM464697,GSM464698,GSM464699,GSM464700,GSM464701,GSM464702,GSM464703,GSM464704,GSM464705,GSM464706,GSM464707,G&quot;| __truncated__ &quot;GSM464658,GSM464659,GSM464660,GSM464661,GSM464662,GSM464663,GSM464664,GSM464665,GSM464666,GSM464667,GSM464668,G&quot;| __truncated__
  .. ..$ sample_organism         : chr &quot;Homo sapiens&quot;
  .. ..$ sample_type             : chr &quot;RNA&quot;
  .. ..$ title                   : chr &quot;Smoking effect on B lymphocytes in females&quot;
  .. ..$ type                    : chr [1:3] &quot;Expression profiling by array&quot; &quot;stress&quot; &quot;stress&quot;
  .. ..$ update_date             : chr &quot;Jul 11 2010&quot;
  .. ..$ value_type              : chr &quot;count&quot;
  .. ..$ web_link                : chr &quot;http://www.ncbi.nlm.nih.gov/geo&quot;</code></pre>
<div id="summary-of-experiment" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.1</span> Summary of experiment<a href="download-data-to-the-current-working-environment.html#summary-of-experiment" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb7-1" title="1"><span class="kw">show</span>(<span class="kw">Meta</span>(gds))</a></code></pre></div>
<pre><code>$channel_count
[1] &quot;1&quot;

$dataset_id
[1] &quot;GDS3713&quot; &quot;GDS3713&quot;

$description
[1] &quot;Analysis of peripheral circulating B cells from smoking and non-smoking healthy US white females. B cells are directly associated with the onset and development of many smoking-induced diseases. Results provide insight into the molecular basis of B cell involvement in smoking-related pathogenesis.&quot;
[2] &quot;control&quot;                                                                                                                                                                                                                                                                                                   
[3] &quot;cigarette smoke&quot;                                                                                                                                                                                                                                                                                           

$email
[1] &quot;geo@ncbi.nlm.nih.gov&quot;

$feature_count
[1] &quot;22283&quot;

$institute
[1] &quot;NCBI NLM NIH&quot;

$name
[1] &quot;Gene Expression Omnibus (GEO)&quot;

$order
[1] &quot;none&quot;

$platform
[1] &quot;GPL96&quot;

$platform_organism
[1] &quot;Homo sapiens&quot;

$platform_technology_type
[1] &quot;in situ oligonucleotide&quot;

$pubmed_id
[1] &quot;20217071&quot;

$ref
[1] &quot;Nucleic Acids Res. 2005 Jan 1;33 Database Issue:D562-6&quot;

$reference_series
[1] &quot;GSE18723&quot;

$sample_count
[1] &quot;79&quot;

$sample_id
[1] &quot;GSM464697,GSM464698,GSM464699,GSM464700,GSM464701,GSM464702,GSM464703,GSM464704,GSM464705,GSM464706,GSM464707,GSM464708,GSM464709,GSM464710,GSM464711,GSM464712,GSM464713,GSM464714,GSM464715,GSM464716,GSM464717,GSM464718,GSM464719,GSM464720,GSM464721,GSM464722,GSM464723,GSM464724,GSM464725,GSM464726,GSM464727,GSM464728,GSM464729,GSM464730,GSM464731,GSM464732,GSM464733,GSM464734,GSM464735,GSM464736&quot;
[2] &quot;GSM464658,GSM464659,GSM464660,GSM464661,GSM464662,GSM464663,GSM464664,GSM464665,GSM464666,GSM464667,GSM464668,GSM464669,GSM464670,GSM464671,GSM464672,GSM464673,GSM464674,GSM464675,GSM464676,GSM464677,GSM464678,GSM464679,GSM464680,GSM464681,GSM464682,GSM464683,GSM464684,GSM464685,GSM464686,GSM464687,GSM464688,GSM464689,GSM464690,GSM464691,GSM464692,GSM464693,GSM464694,GSM464695,GSM464696&quot;          

$sample_organism
[1] &quot;Homo sapiens&quot;

$sample_type
[1] &quot;RNA&quot;

$title
[1] &quot;Smoking effect on B lymphocytes in females&quot;

$type
[1] &quot;Expression profiling by array&quot; &quot;stress&quot;                        &quot;stress&quot;                       

$update_date
[1] &quot;Jul 11 2010&quot;

$value_type
[1] &quot;count&quot;

$web_link
[1] &quot;http://www.ncbi.nlm.nih.gov/geo&quot;</code></pre>
</div>
<div id="assigning-the-expression-and-phenotype-data" class="section level2 hasAnchor">
<h2><span class="header-section-number">4.2</span> Assigning the expression and phenotype data<a href="download-data-to-the-current-working-environment.html#assigning-the-expression-and-phenotype-data" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The expression data is stored in the <code>gds@dataTable@tabl</code> with gene in
The rows and subjects in the columns.
The phenotype data is store at <code>gds@dataTable@columns</code>.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb9-1" title="1">expr &lt;-<span class="st"> </span><span class="kw">Table</span>(gds)</a>
<a class="sourceLine" id="cb9-2" title="2"><span class="kw">summary</span>(expr[, <span class="dv">1</span><span class="op">:</span><span class="dv">4</span>])</a></code></pre></div>
<pre><code>    ID_REF           IDENTIFIER          GSM464697          GSM464698       
 Length:22283       Length:22283       Min.   :    1.53   Min.   :    1.06  
 Class :character   Class :character   1st Qu.:   65.44   1st Qu.:   74.40  
 Mode  :character   Mode  :character   Median :  221.78   Median :  243.15  
                                       Mean   :  859.82   Mean   :  752.89  
                                       3rd Qu.:  582.95   3rd Qu.:  600.85  
                                       Max.   :51356.90   Max.   :34387.10  </code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb11-1" title="1">pheno &lt;-<span class="st"> </span><span class="kw">Columns</span>(gds)</a>
<a class="sourceLine" id="cb11-2" title="2"><span class="kw">summary</span>(pheno)</a></code></pre></div>
<pre><code>    sample                      stress   description       
 Length:79          cigarette smoke:39   Length:79         
 Class :character   control        :40   Class :character  
 Mode  :character                        Mode  :character  </code></pre>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="extracting-data-from-ncbi-gene-expression-omnibus-databse.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="convert-to-gds-to-expression-set.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
