<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Intro | _main.knit</title>
  <meta name="description" content="" />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Intro | _main.knit" />
  <meta property="og:type" content="book" />
  
  
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Intro | _main.knit" />
  
  
  




  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="index.html"/>
<link rel="next" href="extracting-data-from-ncbi-gene-expression-omnibus-databse.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.5.3/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.26/datatables.js"></script>
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>


<style type="text/css">
a.sourceLine { display: inline-block; line-height: 1.25; }
a.sourceLine { pointer-events: none; color: inherit; text-decoration: inherit; }
a.sourceLine:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
a.sourceLine { text-indent: -1em; padding-left: 1em; }
}
pre.numberSource a.sourceLine
  { position: relative; left: -4em; }
pre.numberSource a.sourceLine::before
  { content: attr(title);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; pointer-events: all; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {  }
@media screen {
a.sourceLine::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="1" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i><b>1</b> Preface</a></li>
<li class="chapter" data-level="2" data-path="intro.html"><a href="intro.html"><i class="fa fa-check"></i><b>2</b> Intro</a></li>
<li class="chapter" data-level="3" data-path="extracting-data-from-ncbi-gene-expression-omnibus-databse.html"><a href="extracting-data-from-ncbi-gene-expression-omnibus-databse.html"><i class="fa fa-check"></i><b>3</b> Extracting data from NCBI Gene Expression Omnibus databse</a></li>
<li class="chapter" data-level="4" data-path="download-data-to-the-current-working-environment.html"><a href="download-data-to-the-current-working-environment.html"><i class="fa fa-check"></i><b>4</b> Download data to the current working environment</a><ul>
<li class="chapter" data-level="4.1" data-path="download-data-to-the-current-working-environment.html"><a href="download-data-to-the-current-working-environment.html#summary-of-experiment"><i class="fa fa-check"></i><b>4.1</b> Summary of experiment</a></li>
<li class="chapter" data-level="4.2" data-path="download-data-to-the-current-working-environment.html"><a href="download-data-to-the-current-working-environment.html#assigning-the-expression-and-phenotype-data"><i class="fa fa-check"></i><b>4.2</b> Assigning the expression and phenotype data</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="convert-to-gds-to-expression-set.html"><a href="convert-to-gds-to-expression-set.html"><i class="fa fa-check"></i><b>5</b> convert to GDS to expression set</a></li>
<li class="chapter" data-level="6" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html"><i class="fa fa-check"></i><b>6</b> Query bioloical pathway databases</a></li>
<li class="chapter" data-level="7" data-path="study-dataset.html"><a href="study-dataset.html"><i class="fa fa-check"></i><b>7</b> Study dataset</a></li>
<li class="chapter" data-level="8" data-path="annotation-of-genes.html"><a href="annotation-of-genes.html"><i class="fa fa-check"></i><b>8</b> Annotation of genes</a></li>
<li class="chapter" data-level="9" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html"><i class="fa fa-check"></i><b>9</b> Convert gene symbol to gene ID using <code>biomaRt</code>.</a><ul>
<li class="chapter" data-level="9.1" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html#connect-to-the-selected-biomart-database-using-useensembl-function."><i class="fa fa-check"></i><b>9.1</b> 1. Connect to the selected <em>BioMart</em> database using <code>useEnsembl</code> function.</a><ul>
<li class="chapter" data-level="9.1.1" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html#list-of-available-biomart-databases-hosted-by-ensembl"><i class="fa fa-check"></i><b>9.1.1</b> List of available BioMart databases hosted by Ensembl</a></li>
</ul></li>
<li class="chapter" data-level="9.2" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html#annotate-genes"><i class="fa fa-check"></i><b>9.2</b> 2. Annotate genes</a></li>
<li class="chapter" data-level="9.3" data-path="convert-gene-symbol-to-gene-id-using-biomart..html"><a href="convert-gene-symbol-to-gene-id-using-biomart..html#alternative-methods-for-annotation"><i class="fa fa-check"></i><b>9.3</b> Alternative methods for annotation</a></li>
</ul></li>
<li class="chapter" data-level="10" data-path="curate-biological-functions.html"><a href="curate-biological-functions.html"><i class="fa fa-check"></i><b>10</b> Curate biological functions</a></li>
<li class="chapter" data-level="11" data-path="reference.html"><a href="reference.html"><i class="fa fa-check"></i><b>11</b> Reference</a></li>
<li class="chapter" data-level="12" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html"><i class="fa fa-check"></i><b>12</b> Over-representation Analysis</a></li>
<li class="chapter" data-level="13" data-path="experiment-data.html"><a href="experiment-data.html"><i class="fa fa-check"></i><b>13</b> Experiment Data</a></li>
<li class="chapter" data-level="14" data-path="gene-set-selection.html"><a href="gene-set-selection.html"><i class="fa fa-check"></i><b>14</b> Gene set selection</a><ul>
<li class="chapter" data-level="14.1" data-path="gene-set-selection.html"><a href="gene-set-selection.html#define-differentially-expressed-gene"><i class="fa fa-check"></i><b>14.1</b> Define differentially expressed gene</a></li>
</ul></li>
<li class="chapter" data-level="15" data-path="over-representation-test.html"><a href="over-representation-test.html"><i class="fa fa-check"></i><b>15</b> Over-representation test</a><ul>
<li class="chapter" data-level="15.1" data-path="over-representation-test.html"><a href="over-representation-test.html#gene-set-1"><i class="fa fa-check"></i><b>15.1</b> Gene set 1</a></li>
<li class="chapter" data-level="15.2" data-path="over-representation-test.html"><a href="over-representation-test.html#gene-set-2"><i class="fa fa-check"></i><b>15.2</b> Gene set 2</a></li>
</ul></li>
<li class="chapter" data-level="16" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html"><i class="fa fa-check"></i><b>16</b> Over-respensentation analysis from R package</a><ul>
<li class="chapter" data-level="16.1" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#test-set-up-and-result-interpretations"><i class="fa fa-check"></i><b>16.1</b> Test set up and Result interpretations</a></li>
<li class="chapter" data-level="16.2" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#examples"><i class="fa fa-check"></i><b>16.2</b> Examples</a><ul>
<li class="chapter" data-level="16.2.1" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#go"><i class="fa fa-check"></i><b>16.2.1</b> GO</a></li>
<li class="chapter" data-level="16.2.2" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#kegg"><i class="fa fa-check"></i><b>16.2.2</b> KEGG</a></li>
<li class="chapter" data-level="16.2.3" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#wikipathways"><i class="fa fa-check"></i><b>16.2.3</b> WikiPathways</a></li>
<li class="chapter" data-level="16.2.4" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#reactome"><i class="fa fa-check"></i><b>16.2.4</b> Reactome</a></li>
<li class="chapter" data-level="16.2.5" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#do"><i class="fa fa-check"></i><b>16.2.5</b> DO</a></li>
<li class="chapter" data-level="16.2.6" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#disgenet"><i class="fa fa-check"></i><b>16.2.6</b> DisGeNET</a></li>
<li class="chapter" data-level="16.2.7" data-path="over-respensentation-analysis-from-r-package.html"><a href="over-respensentation-analysis-from-r-package.html#mesh"><i class="fa fa-check"></i><b>16.2.7</b> MeSH</a></li>
</ul></li>
</ul></li>
<li class="chapter" data-level="17" data-path="pros-and-cons-of-ora.html"><a href="pros-and-cons-of-ora.html"><i class="fa fa-check"></i><b>17</b> Pros and Cons of ORA</a></li>
<li class="chapter" data-level="18" data-path="reference-1.html"><a href="reference-1.html"><i class="fa fa-check"></i><b>18</b> Reference</a></li>
<li class="chapter" data-level="19" data-path="gene-set-enrichment-analysis.html"><a href="gene-set-enrichment-analysis.html"><i class="fa fa-check"></i><b>19</b> Gene set enrichment analysis</a></li>
<li class="chapter" data-level="20" data-path="data.html"><a href="data.html"><i class="fa fa-check"></i><b>20</b> Data</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./"></a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="intro" class="section level1 hasAnchor">
<h1><span class="header-section-number">2</span> Intro<a href="intro.html#intro" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<details>
<p><summary>Session Info</summary></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><a class="sourceLine" id="cb1-1" title="1"><span class="kw">sessionInfo</span>()</a></code></pre></div>
<pre><code>## R version 4.1.0 (2021-05-18)
## Platform: x86_64-pc-linux-gnu (64-bit)
## Running under: Ubuntu 20.04.2 LTS
## 
## Matrix products: default
## BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.9.0
## LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.9.0
## 
## locale:
##  [1] LC_CTYPE=C.UTF-8       LC_NUMERIC=C           LC_TIME=C.UTF-8        LC_COLLATE=C.UTF-8    
##  [5] LC_MONETARY=C.UTF-8    LC_MESSAGES=C.UTF-8    LC_PAPER=C.UTF-8       LC_NAME=C             
##  [9] LC_ADDRESS=C           LC_TELEPHONE=C         LC_MEASUREMENT=C.UTF-8 LC_IDENTIFICATION=C   
## 
## attached base packages:
## [1] parallel  stats4    stats     graphics  grDevices utils     datasets  methods   base     
## 
## other attached packages:
##  [1] GEOquery_2.60.0       dplyr_1.0.10          limma_3.48.2          DT_0.26              
##  [5] biomaRt_2.48.2        org.Hs.eg.db_3.13.0   AnnotationDbi_1.54.1  IRanges_2.26.0       
##  [9] S4Vectors_0.30.0      Biobase_2.52.0        BiocGenerics_0.38.0   DOSE_3.18.3          
## [13] enrichplot_1.12.3     clusterProfiler_4.0.5 here_1.0.1           
## 
## loaded via a namespace (and not attached):
##   [1] shadowtext_0.1.1       fastmatch_1.1-3        servr_0.22             BiocFileCache_2.0.0   
##   [5] plyr_1.8.6             igraph_1.2.6           lazyeval_0.2.2         splines_4.1.0         
##   [9] BiocParallel_1.26.1    crosstalk_1.1.1        GenomeInfoDb_1.28.1    ggplot2_3.4.0         
##  [13] digest_0.6.27          yulab.utils_0.0.4      htmltools_0.5.5        GOSemSim_2.18.1       
##  [17] viridis_0.6.1          GO.db_3.13.0           fansi_0.5.0            magrittr_2.0.3        
##  [21] memoise_2.0.1          scriptResume_0.1.4     tzdb_0.1.2             readr_2.0.0           
##  [25] Biostrings_2.60.1      graphlayouts_0.8.0     vroom_1.5.3            prettyunits_1.1.1     
##  [29] colorspace_2.0-2       blob_1.2.2             rappdirs_0.3.3         ggrepel_0.9.1         
##  [33] xfun_0.39              callr_3.7.3            crayon_1.4.1           RCurl_1.98-1.3        
##  [37] jsonlite_1.8.4         scatterpie_0.1.7       ape_5.6-2              glue_1.6.2            
##  [41] polyclip_1.10-0        gtable_0.3.0           zlibbioc_1.38.0        XVector_0.32.0        
##  [45] webshot_0.5.2          scales_1.2.1           DBI_1.1.1              Rcpp_1.0.7            
##  [49] viridisLite_0.4.0      progress_1.2.2         gridGraphics_0.5-1     tidytree_0.3.9        
##  [53] bit_4.0.4              htmlwidgets_1.5.3      httr_1.4.2             fgsea_1.18.0          
##  [57] RColorBrewer_1.1-2     ellipsis_0.3.2         pkgconfig_2.0.3        XML_3.99-0.6          
##  [61] farver_2.1.0           sass_0.4.0             dbplyr_2.1.1           utf8_1.2.2            
##  [65] later_1.2.0            ggplotify_0.1.0        tidyselect_1.1.1       rlang_1.1.1           
##  [69] reshape2_1.4.4         munsell_0.5.0          tools_4.1.0            cachem_1.0.5          
##  [73] downloader_0.4         cli_3.4.1              generics_0.1.0         RSQLite_2.2.7         
##  [77] evaluate_0.15          stringr_1.5.0          fastmap_1.1.0          yaml_2.2.1            
##  [81] ggtree_3.0.4           processx_3.8.1         knitr_1.39             bit64_4.0.5           
##  [85] tidygraph_1.2.0        purrr_0.3.4            KEGGREST_1.32.0        ggraph_2.0.5          
##  [89] nlme_3.1-152           mime_0.11              aplot_0.1.2            DO.db_2.9             
##  [93] xml2_1.3.2             compiler_4.1.0         rstudioapi_0.13        filelock_1.0.2        
##  [97] curl_4.3.2             png_0.1-7              treeio_1.16.2          statmod_1.4.36        
## [101] tibble_3.1.6           tweenr_1.0.2           bslib_0.4.2            stringi_1.7.8         
## [105] ps_1.6.0               highr_0.9              lattice_0.20-44        Matrix_1.5-3          
## [109] markdown_1.1           vctrs_0.5.1            pillar_1.6.2           lifecycle_1.0.3       
## [113] jquerylib_0.1.4        data.table_1.14.0      cowplot_1.1.1          bitops_1.0-7          
## [117] httpuv_1.6.1           patchwork_1.1.1        qvalue_2.24.0          R6_2.5.1              
## [121] promises_1.2.0.1       bookdown_0.34          gridExtra_2.3          MASS_7.3-54           
## [125] assertthat_0.2.1       rprojroot_2.0.2        withr_2.5.0            GenomeInfoDbData_1.2.6
## [129] hms_1.1.0              grid_4.1.0             ggfun_0.0.5            tidyr_1.2.1           
## [133] rmarkdown_2.18         ggforce_0.3.3</code></pre>
</details>

</div>
            </section>

          </div>
        </div>
      </div>
<a href="index.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="extracting-data-from-ncbi-gene-expression-omnibus-databse.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
