<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>4 Query bioloical pathway databases | Pathway analysis methods</title>
  <meta name="description" content="This notebook is dedicated to genetic pathway analysis methods." />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="4 Query bioloical pathway databases | Pathway analysis methods" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This notebook is dedicated to genetic pathway analysis methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="4 Query bioloical pathway databases | Pathway analysis methods" />
  
  <meta name="twitter:description" content="This notebook is dedicated to genetic pathway analysis methods." />
  

<meta name="author" content="Siying Huang (E0482362)" />


<meta name="date" content="2024-01-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="differential-gene-expression-analysis-dgea.html"/>
<link rel="next" href="over-representation-analysis.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.26/datatables.js"></script>
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Prefeace</a></li>
<li class="chapter" data-level="1" data-path="gene-sets-and-pathways.html"><a href="gene-sets-and-pathways.html"><i class="fa fa-check"></i><b>1</b> Gene sets and pathways</a>
<ul>
<li class="chapter" data-level="1.1" data-path="gene-sets-and-pathways.html"><a href="gene-sets-and-pathways.html#organization-of-this-e-book"><i class="fa fa-check"></i><b>1.1</b> Organization of this e-book</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="study-dataset.html"><a href="study-dataset.html"><i class="fa fa-check"></i><b>2</b> Study dataset</a>
<ul>
<li class="chapter" data-level="2.1" data-path="study-dataset.html"><a href="study-dataset.html#extracting-data-from-ncbi-gene-expression-omnibus-databse"><i class="fa fa-check"></i><b>2.1</b> Extracting data from NCBI Gene Expression Omnibus databse</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="study-dataset.html"><a href="study-dataset.html#summary-of-the-study"><i class="fa fa-check"></i><b>2.1.1</b> Summary of the study</a></li>
<li class="chapter" data-level="2.1.2" data-path="study-dataset.html"><a href="study-dataset.html#assigning-the-expression-and-phenotype-data"><i class="fa fa-check"></i><b>2.1.2</b> Assigning the expression and phenotype data</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="study-dataset.html"><a href="study-dataset.html#convert-to-gds-to-expression-set"><i class="fa fa-check"></i><b>2.2</b> convert to GDS to expression set</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-gene-expression-analysis-dgea.html"><a href="differential-gene-expression-analysis-dgea.html"><i class="fa fa-check"></i><b>3</b> Differential gene expression analysis (DGEA)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-gene-expression-analysis-dgea.html"><a href="differential-gene-expression-analysis-dgea.html#comparison-between-smokers-and-non-smokers"><i class="fa fa-check"></i><b>3.1</b> Comparison between smokers and non-smokers</a></li>
<li class="chapter" data-level="3.2" data-path="differential-gene-expression-analysis-dgea.html"><a href="differential-gene-expression-analysis-dgea.html#results-visualization"><i class="fa fa-check"></i><b>3.2</b> Results visualization</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html"><i class="fa fa-check"></i><b>4</b> Query bioloical pathway databases</a>
<ul>
<li class="chapter" data-level="4.1" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#gene-annotation"><i class="fa fa-check"></i><b>4.1</b> Gene annotation</a></li>
<li class="chapter" data-level="4.2" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#convert-gene-symbol-to-gene-id-using-biomart."><i class="fa fa-check"></i><b>4.2</b> Convert gene symbol to gene ID using <code>biomaRt</code>.</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#connect-to-the-selected-biomart-database-using-useensembl-function."><i class="fa fa-check"></i><b>4.2.1</b> Connect to the selected <em>BioMart</em> database using <code>useEnsembl</code> function.</a></li>
<li class="chapter" data-level="4.2.2" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#annotate-genes"><i class="fa fa-check"></i><b>4.2.2</b> Annotate genes</a></li>
<li class="chapter" data-level="4.2.3" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#alternative-methods-for-annotation"><i class="fa fa-check"></i><b>4.2.3</b> Alternative methods for annotation</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#curate-biological-functions"><i class="fa fa-check"></i><b>4.3</b> Curate biological functions</a></li>
<li class="chapter" data-level="4.4" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#reference"><i class="fa fa-check"></i><b>4.4</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html"><i class="fa fa-check"></i><b>5</b> Over-representation Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#data-prepartion"><i class="fa fa-check"></i><b>5.1</b> Data prepartion</a></li>
<li class="chapter" data-level="5.2" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#gene-set-selection"><i class="fa fa-check"></i><b>5.2</b> Gene set selection</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#define-differentially-expressed-gene"><i class="fa fa-check"></i><b>5.2.1</b> Define differentially expressed gene</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#over-representation-analysis-1"><i class="fa fa-check"></i><b>5.3</b> Over-representation analysis</a></li>
<li class="chapter" data-level="5.4" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#over-respensentation-analysis-from-r-package"><i class="fa fa-check"></i><b>5.4</b> Over-respensentation analysis from R package</a></li>
<li class="chapter" data-level="5.5" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#test-set-up-and-result-interpretations"><i class="fa fa-check"></i><b>5.5</b> Test set up and result interpretations</a></li>
<li class="chapter" data-level="5.6" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#examples"><i class="fa fa-check"></i><b>5.6</b> Examples</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#go"><i class="fa fa-check"></i><b>5.6.1</b> GO</a></li>
<li class="chapter" data-level="5.6.2" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#kegg"><i class="fa fa-check"></i><b>5.6.2</b> KEGG</a></li>
<li class="chapter" data-level="5.6.3" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#wikipathways"><i class="fa fa-check"></i><b>5.6.3</b> WikiPathways</a></li>
<li class="chapter" data-level="5.6.4" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#reactome"><i class="fa fa-check"></i><b>5.6.4</b> Reactome</a></li>
<li class="chapter" data-level="5.6.5" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#do"><i class="fa fa-check"></i><b>5.6.5</b> DO</a></li>
<li class="chapter" data-level="5.6.6" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#disgenet"><i class="fa fa-check"></i><b>5.6.6</b> DisGeNET</a></li>
<li class="chapter" data-level="5.6.7" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#mesh"><i class="fa fa-check"></i><b>5.6.7</b> MeSH</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#pros-and-cons-of-ora"><i class="fa fa-check"></i><b>5.7</b> Pros and Cons of ORA</a></li>
<li class="chapter" data-level="5.8" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#reference-1"><i class="fa fa-check"></i><b>5.8</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i>Appendix</a>
<ul>
<li class="chapter" data-level="5.9" data-path="appendix.html"><a href="appendix.html#biological-process-data-bases"><i class="fa fa-check"></i><b>5.9</b> Biological process data bases</a></li>
<li class="chapter" data-level="5.10" data-path="appendix.html"><a href="appendix.html#summary-of-currently-available-pathway-analysis-methods"><i class="fa fa-check"></i><b>5.10</b> Summary of currently available pathway analysis methods</a>
<ul>
<li class="chapter" data-level="5.10.1" data-path="appendix.html"><a href="appendix.html#over-representation-analysis-2"><i class="fa fa-check"></i><b>5.10.1</b> Over-representation analysis</a></li>
<li class="chapter" data-level="5.10.2" data-path="appendix.html"><a href="appendix.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>5.10.2</b> Gene set enrichment analysis</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pathway analysis methods</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="query-bioloical-pathway-databases" class="section level1 hasAnchor" number="4">
<h1><span class="header-section-number">4</span> Query bioloical pathway databases<a href="query-bioloical-pathway-databases.html#query-bioloical-pathway-databases" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>For this practice, we’ll use the result from differential gene expression
analysis.</p>
<div id="gene-annotation" class="section level2 hasAnchor" number="4.1">
<h2><span class="header-section-number">4.1</span> Gene annotation<a href="query-bioloical-pathway-databases.html#gene-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>The GDS3713 dataset comes with an annotation file that has information
for genes in the array:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="query-bioloical-pathway-databases.html#cb18-1" aria-hidden="true" tabindex="-1"></a><span class="fu">cat</span>(<span class="fu">paste0</span>(<span class="fu">names</span>(annot), <span class="at">collapse =</span> <span class="st">&#39;</span><span class="sc">\t</span><span class="st">&#39;</span>))</span></code></pre></div>
<pre><code>ID  Gene.title  Gene.symbol Gene.ID UniGene.title   UniGene.symbol  UniGene.ID  Nucleotide.Title    GI  GenBank.Accession   Platform_CLONEID    Platform_ORF    Platform_SPOTID Chromosome.location Chromosome.annotation   GO.Function GO.Process  GO.Component    GO.Function.ID  GO.Process.ID   GO.Component.ID</code></pre>
<p>One can directly query a specific gene from this file. For example, we can
look up the ICOSLG information.</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="query-bioloical-pathway-databases.html#cb20-1" aria-hidden="true" tabindex="-1"></a><span class="fu">datatable</span>(annot[<span class="fu">which</span>(annot<span class="sc">$</span>Gene.symbol <span class="sc">==</span> <span class="st">&quot;ICOSLG&quot;</span>), ],</span>
<span id="cb20-2"><a href="query-bioloical-pathway-databases.html#cb20-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">rownames =</span> F</span>
<span id="cb20-3"><a href="query-bioloical-pathway-databases.html#cb20-3" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-330ed6dad6c0a38f0a14" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-330ed6dad6c0a38f0a14">{"x":{"filter":"none","vertical":false,"data":[["211197_s_at","211198_s_at","211199_s_at","213450_s_at"],["inducible T-cell costimulator ligand","inducible T-cell costimulator ligand","inducible T-cell costimulator ligand","inducible T-cell costimulator ligand"],["ICOSLG","ICOSLG","ICOSLG","ICOSLG"],["23308","23308","23308","23308"],[null,null,null,null],[null,null,null,null],[null,null,null,null],["Homo sapiens EST from clone 34465, full insert","Homo sapiens transmembrane protein B7-H2 ICOS ligand mRNA, complete cds","Homo sapiens B7-like protein (GL50) mRNA, complete cds",null],[7799138,9858866,6983943,null],["AL355690","AF289028","AF199028",null],[null,null,null,null],[null,null,null,null],[null,null,null,null],["21q22.3","21q22.3","21q22.3","21q22.3"],["Chromosome 21, NC_000021.9 (44216980..44241447, complement)","Chromosome 21, NC_000021.9 (44216980..44241447, complement)","Chromosome 21, NC_000021.9 (44216980..44241447, complement)","Chromosome 21, NC_000021.9 (44216980..44241447, complement)"],["receptor binding","receptor binding","receptor binding","receptor binding"],["B cell activation///T cell activation///T cell costimulation///adaptive immune response///defense response///hyperosmotic response///positive regulation of activated T cell proliferation///signal transduction","B cell activation///T cell activation///T cell costimulation///adaptive immune response///defense response///hyperosmotic response///positive regulation of activated T cell proliferation///signal transduction","B cell activation///T cell activation///T cell costimulation///adaptive immune response///defense response///hyperosmotic response///positive regulation of activated T cell proliferation///signal transduction","B cell activation///T cell activation///T cell costimulation///adaptive immune response///defense response///hyperosmotic response///positive regulation of activated T cell proliferation///signal transduction"],["extracellular exosome///integral component of membrane///plasma membrane","extracellular exosome///integral component of membrane///plasma membrane","extracellular exosome///integral component of membrane///plasma membrane","extracellular exosome///integral component of membrane///plasma membrane"],["GO:0005102","GO:0005102","GO:0005102","GO:0005102"],["GO:0042113///GO:0042110///GO:0031295///GO:0002250///GO:0006952///GO:0006972///GO:0042104///GO:0007165","GO:0042113///GO:0042110///GO:0031295///GO:0002250///GO:0006952///GO:0006972///GO:0042104///GO:0007165","GO:0042113///GO:0042110///GO:0031295///GO:0002250///GO:0006952///GO:0006972///GO:0042104///GO:0007165","GO:0042113///GO:0042110///GO:0031295///GO:0002250///GO:0006952///GO:0006972///GO:0042104///GO:0007165"],["GO:0070062///GO:0016021///GO:0005886","GO:0070062///GO:0016021///GO:0005886","GO:0070062///GO:0016021///GO:0005886","GO:0070062///GO:0016021///GO:0005886"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th>ID<\/th>\n      <th>Gene.title<\/th>\n      <th>Gene.symbol<\/th>\n      <th>Gene.ID<\/th>\n      <th>UniGene.title<\/th>\n      <th>UniGene.symbol<\/th>\n      <th>UniGene.ID<\/th>\n      <th>Nucleotide.Title<\/th>\n      <th>GI<\/th>\n      <th>GenBank.Accession<\/th>\n      <th>Platform_CLONEID<\/th>\n      <th>Platform_ORF<\/th>\n      <th>Platform_SPOTID<\/th>\n      <th>Chromosome.location<\/th>\n      <th>Chromosome.annotation<\/th>\n      <th>GO.Function<\/th>\n      <th>GO.Process<\/th>\n      <th>GO.Component<\/th>\n      <th>GO.Function.ID<\/th>\n      <th>GO.Process.ID<\/th>\n      <th>GO.Component.ID<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":8}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>However, most of time, we don’t have such a file that comes with the
trancriptomic data. We will then need to annotate the genes.</p>
</div>
<div id="convert-gene-symbol-to-gene-id-using-biomart." class="section level2 hasAnchor" number="4.2">
<h2><span class="header-section-number">4.2</span> Convert gene symbol to gene ID using <code>biomaRt</code>.<a href="query-bioloical-pathway-databases.html#convert-gene-symbol-to-gene-id-using-biomart." class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Since many pathway database requires entrez gene IDs as entries and most of
sequencing/ microarray output provides gene symbols or ensembl IDs, we’ll
need to convert gene symbol to entrez IDs for the pathway analysis.</p>
<p>We can query from the <a href="http://www.ensembl.org/index.html">ensembl</a> site manually.
Or we can do better using<code>biomaRt</code>package.
For a full tutorial of <code>biomaRt</code> package, see <a href="https://bioconductor.org/packages/release/bioc/vignettes/biomaRt/inst/doc/accessing_ensembl.html">here</a>. It covers
some more advanced use case. Here I only cover basic usage of this package.</p>
<div id="connect-to-the-selected-biomart-database-using-useensembl-function." class="section level3 hasAnchor" number="4.2.1">
<h3><span class="header-section-number">4.2.1</span> Connect to the selected <em>BioMart</em> database using <code>useEnsembl</code> function.<a href="query-bioloical-pathway-databases.html#connect-to-the-selected-biomart-database-using-useensembl-function." class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Before connecting to a specific database, we can first check available
databases.</p>
<div id="list-of-available-biomart-databases-hosted-by-ensembl" class="section level4 hasAnchor" number="4.2.1.1">
<h4><span class="header-section-number">4.2.1.1</span> List of available BioMart databases hosted by Ensembl<a href="query-bioloical-pathway-databases.html#list-of-available-biomart-databases-hosted-by-ensembl" class="anchor-section" aria-label="Anchor link to header"></a></h4>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="query-bioloical-pathway-databases.html#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># available database</span></span>
<span id="cb21-2"><a href="query-bioloical-pathway-databases.html#cb21-2" aria-hidden="true" tabindex="-1"></a><span class="fu">datatable</span>(<span class="fu">listEnsembl</span>())</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-492b7729bf2e91d55007" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-492b7729bf2e91d55007">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4"],["genes","mouse_strains","snps","regulation"],["Ensembl Genes 111","Mouse strains 111","Ensembl Variation 111","Ensembl Regulation 111"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>biomart<\/th>\n      <th>version<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="query-bioloical-pathway-databases.html#cb22-1" aria-hidden="true" tabindex="-1"></a><span class="co"># available biological species for the gene databases</span></span>
<span id="cb22-2"><a href="query-bioloical-pathway-databases.html#cb22-2" aria-hidden="true" tabindex="-1"></a>mart <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(<span class="st">&quot;genes&quot;</span>, <span class="at">mirror =</span> <span class="st">&#39;www&#39;</span>)</span>
<span id="cb22-3"><a href="query-bioloical-pathway-databases.html#cb22-3" aria-hidden="true" tabindex="-1"></a><span class="fu">datatable</span>(<span class="fu">listDatasets</span>(mart))</span></code></pre></div>
<div class="datatables html-widget html-fill-item-overflow-hidden html-fill-item" id="htmlwidget-6dd1f603825f3c8ebd2b" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-6dd1f603825f3c8ebd2b">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94","95","96","97","98","99","100","101","102","103","104","105","106","107","108","109","110","111","112","113","114","115","116","117","118","119","120","121","122","123","124","125","126","127","128","129","130","131","132","133","134","135","136","137","138","139","140","141","142","143","144","145","146","147","148","149","150","151","152","153","154","155","156","157","158","159","160","161","162","163","164","165","166","167","168","169","170","171","172","173","174","175","176","177","178","179","180","181","182","183","184","185","186","187","188","189","190","191","192","193","194","195","196","197","198","199","200","201","202","203","204","205","206","207","208","209","210","211","212","213","214"],["abrachyrhynchus_gene_ensembl","acalliptera_gene_ensembl","acarolinensis_gene_ensembl","acchrysaetos_gene_ensembl","acitrinellus_gene_ensembl","amelanoleuca_gene_ensembl","amexicanus_gene_ensembl","anancymaae_gene_ensembl","aocellaris_gene_ensembl","apercula_gene_ensembl","aplatyrhynchos_gene_ensembl","apolyacanthus_gene_ensembl","applatyrhynchos_gene_ensembl","atestudineus_gene_ensembl","bbbison_gene_ensembl","bgrunniens_gene_ensembl","bihybrid_gene_ensembl","bmusculus_gene_ensembl","bmutus_gene_ensembl","bsplendens_gene_ensembl","btaurus_gene_ensembl","cabingdonii_gene_ensembl","catys_gene_ensembl","cauratus_gene_ensembl","cccarpio_gene_ensembl","cdromedarius_gene_ensembl","celegans_gene_ensembl","cgchok1gshd_gene_ensembl","cgobio_gene_ensembl","charengus_gene_ensembl","chircus_gene_ensembl","choffmanni_gene_ensembl","chyarkandensis_gene_ensembl","cimitator_gene_ensembl","cintestinalis_gene_ensembl","cjacchus_gene_ensembl","cjaponica_gene_ensembl","clanigera_gene_ensembl","cldingo_gene_ensembl","clfamiliaris_gene_ensembl","clumpus_gene_ensembl","cmilii_gene_ensembl","cpbellii_gene_ensembl","cporcellus_gene_ensembl","cporosus_gene_ensembl","csabaeus_gene_ensembl","csavignyi_gene_ensembl","csemilaevis_gene_ensembl","csyrichta_gene_ensembl","cvariegatus_gene_ensembl","cwagneri_gene_ensembl","dclupeoides_gene_ensembl","dlabrax_gene_ensembl","dleucas_gene_ensembl","dmelanogaster_gene_ensembl","dnovemcinctus_gene_ensembl","dordii_gene_ensembl","drerio_gene_ensembl","easinus_gene_ensembl","eburgeri_gene_ensembl","ecaballus_gene_ensembl","ecalabaricus_gene_ensembl","eelectricus_gene_ensembl","eeuropaeus_gene_ensembl","elucius_gene_ensembl","etelfairi_gene_ensembl","falbicollis_gene_ensembl","fcatus_gene_ensembl","fheteroclitus_gene_ensembl","gaculeatus_gene_ensembl","gevgoodei_gene_ensembl","gfortis_gene_ensembl","ggallus_gene_ensembl","ggorilla_gene_ensembl","gmorhua_gene_ensembl","hburtoni_gene_ensembl","hcomes_gene_ensembl","hgfemale_gene_ensembl","hhucho_gene_ensembl","hsapiens_gene_ensembl","ipunctatus_gene_ensembl","itridecemlineatus_gene_ensembl","jjaculus_gene_ensembl","kmarmoratus_gene_ensembl","lafricana_gene_ensembl","lbergylta_gene_ensembl","lcalcarifer_gene_ensembl","lchalumnae_gene_ensembl","lcrocea_gene_ensembl","llaticaudata_gene_ensembl","lleishanense_gene_ensembl","loculatus_gene_ensembl","marmatus_gene_ensembl","mauratus_gene_ensembl","mcaroli_gene_ensembl","mdomestica_gene_ensembl","mfascicularis_gene_ensembl","mgallopavo_gene_ensembl","mleucophaeus_gene_ensembl","mlucifugus_gene_ensembl","mmmarmota_gene_ensembl","mmonoceros_gene_ensembl","mmoschiferus_gene_ensembl","mmulatta_gene_ensembl","mmurdjan_gene_ensembl","mmurinus_gene_ensembl","mmusculus_gene_ensembl","mnemestrina_gene_ensembl","mochrogaster_gene_ensembl","mpahari_gene_ensembl","mpfuro_gene_ensembl","mspicilegus_gene_ensembl","mspretus_gene_ensembl","mzebra_gene_ensembl","nbrichardi_gene_ensembl","neugenii_gene_ensembl","nfurzeri_gene_ensembl","ngalili_gene_ensembl","nleucogenys_gene_ensembl","nnaja_gene_ensembl","nscutatus_gene_ensembl","nvison_gene_ensembl","oanatinus_gene_ensembl","oarambouillet_gene_ensembl","oaries_gene_ensembl","ocuniculus_gene_ensembl","odegus_gene_ensembl","ogarnettii_gene_ensembl","ojavanicus_gene_ensembl","okisutch_gene_ensembl","olatipes_gene_ensembl","omelastigma_gene_ensembl","omykiss_gene_ensembl","oniloticus_gene_ensembl","oprinceps_gene_ensembl","osinensis_gene_ensembl","otshawytscha_gene_ensembl","pabelii_gene_ensembl","panubis_gene_ensembl","pcapensis_gene_ensembl","pcatodon_gene_ensembl","pcinereus_gene_ensembl","pcoquereli_gene_ensembl","pformosa_gene_ensembl","pkingsleyae_gene_ensembl","platipinna_gene_ensembl","pleo_gene_ensembl","pmajor_gene_ensembl","pmarinus_gene_ensembl","pmbairdii_gene_ensembl","pmuralis_gene_ensembl","pnattereri_gene_ensembl","pnyererei_gene_ensembl","ppaniscus_gene_ensembl","ppardus_gene_ensembl","preticulata_gene_ensembl","psimus_gene_ensembl","psinensis_gene_ensembl","psinus_gene_ensembl","ptaltaica_gene_ensembl","ptextilis_gene_ensembl","ptroglodytes_gene_ensembl","pvampyrus_gene_ensembl","rbieti_gene_ensembl","rferrumequinum_gene_ensembl","rnorvegicus_gene_ensembl","rroxellana_gene_ensembl","saraneus_gene_ensembl","saurata_gene_ensembl","sbboliviensis_gene_ensembl","scanaria_gene_ensembl","scaustralis_gene_ensembl","scerevisiae_gene_ensembl","sdumerili_gene_ensembl","sformosus_gene_ensembl","shabroptila_gene_ensembl","sharrisii_gene_ensembl","sldorsalis_gene_ensembl","slucioperca_gene_ensembl","smaximus_gene_ensembl","smerianae_gene_ensembl","spartitus_gene_ensembl","spunctatus_gene_ensembl","ssalar_gene_ensembl","ssbamei_gene_ensembl","ssberkshire_gene_ensembl","sscrofa_gene_ensembl","sshampshire_gene_ensembl","ssjinhua_gene_ensembl","sslandrace_gene_ensembl","sslargewhite_gene_ensembl","ssmeishan_gene_ensembl","sspietrain_gene_ensembl","ssrongchang_gene_ensembl","sstibetan_gene_ensembl","ssusmarc_gene_ensembl","sswuzhishan_gene_ensembl","strutta_gene_ensembl","svulgaris_gene_ensembl","tbelangeri_gene_ensembl","tctriunguis_gene_ensembl","tguttata_gene_ensembl","tnigroviridis_gene_ensembl","trubripes_gene_ensembl","ttruncatus_gene_ensembl","uamericanus_gene_ensembl","umaritimus_gene_ensembl","uparryii_gene_ensembl","vpacos_gene_ensembl","vursinus_gene_ensembl","vvulpes_gene_ensembl","xcouchianus_gene_ensembl","xmaculatus_gene_ensembl","xtropicalis_gene_ensembl"],["Pink-footed goose genes (ASM259213v1)","Eastern happy genes (fAstCal1.2)","Green anole genes (AnoCar2.0v2)","Golden eagle genes (bAquChr1.2)","Midas cichlid genes (Midas_v5)","Giant panda genes (ASM200744v2)","Mexican tetra genes (Astyanax_mexicanus-2.0)","Ma's night monkey genes (Anan_2.0)","Clown anemonefish genes (AmpOce1.0)","Orange clownfish genes (Nemo_v1)","Mallard genes (ASM874695v1)","Spiny chromis genes (ASM210954v1)","Duck genes (CAU_duck1.0)","Climbing perch genes (fAnaTes1.2)","American bison genes (Bison_UMD1.0)","Domestic yak genes (LU_Bosgru_v3.0)","Hybrid - Bos Indicus genes (UOA_Brahman_1)","Blue whale genes (mBalMus1.v2)","Wild yak genes (BosGru_v2.0)","Siamese fighting fish genes (fBetSpl5.2)","Cow genes (ARS-UCD1.3)","Abingdon island giant tortoise genes (ASM359739v1)","Sooty mangabey genes (Caty_1.0)","Goldfish genes (ASM336829v1)","Common carp genes (Cypcar_WagV4.0)","Arabian camel genes (CamDro2)","Caenorhabditis elegans (Nematode, N2) genes (WBcel235)","Chinese hamster CHOK1GS genes (CHOK1GS_HDv1)","Channel bull blenny genes (fCotGob3.1)","Atlantic herring genes (Ch_v2.0.2)","Goat genes (ARS1)","Sloth genes (choHof1)","Yarkand deer genes (CEY_v1)","Panamanian white-faced capuchin genes (Cebus_imitator-1.0)","C.intestinalis genes (KH)","White-tufted-ear marmoset genes (mCalJac1.pat.X)","Japanese quail genes (Coturnix_japonica_2.0)","Long-tailed chinchilla genes (ChiLan1.0)","Dingo genes (ASM325472v1)","Dog genes (ROS_Cfam_1.0)","Lumpfish genes (fCycLum1.pri)","Elephant shark genes (Callorhinchus_milii-6.1.3)","Painted turtle genes (Chrysemys_picta_bellii-3.0.3)","Guinea Pig genes (Cavpor3.0)","Australian saltwater crocodile genes (CroPor_comp1)","Vervet-AGM genes (ChlSab1.1)","C.savignyi genes (CSAV 2.0)","Tongue sole genes (Cse_v1.0)","Tarsier genes (Tarsius_syrichta-2.0.1)","Sheepshead minnow genes (C_variegatus-1.0)","Chacoan peccary genes (CatWag_v2_BIUU_UCD)","Denticle herring genes (fDenClu1.1)","European seabass genes (dlabrax2021)","Beluga whale genes (ASM228892v3)","Drosophila melanogaster (Fruit fly) genes (BDGP6.46)","Armadillo genes (Dasnov3.0)","Kangaroo rat genes (Dord_2.0)","Zebrafish genes (GRCz11)","Donkey genes (ASM1607732v2)","Hagfish genes (Eburgeri_3.2)","Horse genes (EquCab3.0)","Reedfish genes (fErpCal1.1)","Electric eel genes (Ee_SOAP_WITH_SSPACE)","Hedgehog genes (eriEur1)","Northern pike genes (Eluc_v4)","Lesser hedgehog tenrec genes (TENREC)","Collared flycatcher genes (FicAlb1.5)","Cat genes (Felis_catus_9.0)","Mummichog genes (Fundulus_heteroclitus-3.0.2)","Stickleback genes (BROAD S1)","Goodes thornscrub tortoise genes (rGopEvg1_v1.p)","Medium ground-finch genes (GeoFor_1.0)","Chicken genes (bGalGal1.mat.broiler.GRCg7b)","Gorilla genes (gorGor4)","Atlantic cod genes (gadMor3.0)","Burton's mouthbrooder genes (AstBur1.0)","Tiger tail seahorse genes (H_comes_QL1_v1)","Naked mole-rat female genes (Naked_mole-rat_maternal)","Huchen genes (ASM331708v1)","Human genes (GRCh38.p14)","Channel catfish genes (IpCoco_1.2)","Squirrel genes (SpeTri2.0)","Lesser Egyptian jerboa genes (JacJac1.0)","Mangrove rivulus genes (ASM164957v1)","Elephant genes (Loxafr3.0)","Ballan wrasse genes (BallGen_V1)","Barramundi perch genes (ASB_HGAPassembly_v1)","Coelacanth genes (LatCha1)","Large yellow croaker genes (L_crocea_2.0)","Blue-ringed sea krait genes (latLat_1.0)","Leishan spiny toad genes (ASM966780v1)","Spotted gar genes (LepOcu1)","Zig-zag eel genes (fMasArm1.2)","Golden Hamster genes (MesAur1.0)","Ryukyu mouse genes (CAROLI_EIJ_v1.1)","Opossum genes (ASM229v1)","Crab-eating macaque genes (Macaca_fascicularis_6.0)","Turkey genes (Turkey_5.1)","Drill genes (Mleu.le_1.0)","Microbat genes (Myoluc2.0)","Alpine marmot genes (marMar2.1)","Narwhal genes (NGI_Narwhal_1)","Siberian musk deer genes (MosMos_v2_BIUU_UCD)","Macaque genes (Mmul_10)","Pinecone soldierfish genes (fMyrMur1.1)","Mouse Lemur genes (Mmur_3.0)","Mouse genes (GRCm39)","Pig-tailed macaque genes (Mnem_1.0)","Prairie vole genes (MicOch1.0)","Shrew mouse genes (PAHARI_EIJ_v1.1)","Ferret genes (MusPutFur1.0)","Steppe mouse genes (MUSP714)","Algerian mouse genes (SPRET_EiJ_v1)","Zebra mbuna genes (M_zebra_UMD2a)","Lyretail cichlid genes (NeoBri1.0)","Wallaby genes (Meug_1.0)","Turquoise killifish genes (Nfu_20140520)","Upper Galilee mountains blind mole rat genes (S.galili_v1.0)","Gibbon genes (Nleu_3.0)","Indian cobra genes (Nana_v5)","Mainland tiger snake genes (TS10Xv2-PRI)","American mink genes (NNQGG.v01)","Platypus genes (mOrnAna1.p.v1)","Sheep genes (ARS-UI_Ramb_v2.0)","Sheep (texel) genes (Oar_v3.1)","Rabbit genes (OryCun2.0)","Degu genes (OctDeg1.0)","Bushbaby genes (OtoGar3)","Javanese ricefish genes (OJAV_1.1)","Coho salmon genes (Okis_V2)","Japanese medaka HdrR genes (ASM223467v1)","Indian medaka genes (Om_v0.7.RACA)","Rainbow trout genes (USDA_OmykA_1.1)","Nile tilapia genes (O_niloticus_UMD_NMBU)","Pika genes (OchPri2.0-Ens)","Chinese medaka genes (ASM858656v1)","Chinook salmon genes (Otsh_v1.0)","Sumatran orangutan genes (Susie_PABv2)","Olive baboon genes (Panubis1.0)","Hyrax genes (proCap1)","Sperm whale genes (ASM283717v2)","Koala genes (phaCin_unsw_v4.1)","Coquerel's sifaka genes (Pcoq_1.0)","Amazon molly genes (Poecilia_formosa-5.1.2)","Paramormyrops kingsleyae genes (PKINGS_0.1)","Sailfin molly genes (P_latipinna-1.0)","Lion genes (PanLeo1.0)","Great Tit genes (Parus_major1.1)","Lamprey genes (Pmarinus_7.0)","Northern American deer mouse genes (HU_Pman_2.1)","Common wall lizard genes (PodMur_1.0)","Red-bellied piranha genes (Pygocentrus_nattereri-1.0.2)","Makobe Island cichlid genes (PunNye1.0)","Bonobo genes (panpan1.1)","Leopard genes (PanPar1.0)","Guppy genes (Guppy_female_1.0_MT)","Greater bamboo lemur genes (Prosim_1.0)","Chinese softshell turtle genes (PelSin_1.0)","Vaquita genes (mPhoSin1.pri)","Tiger genes (PanTig1.0)","Eastern brown snake genes (EBS10Xv2-PRI)","Chimpanzee genes (Pan_tro_3.0)","Megabat genes (pteVam1)","Black snub-nosed monkey genes (ASM169854v1)","Greater horseshoe bat genes (mRhiFer1_v1.p)","Rat genes (mRatBN7.2)","Golden snub-nosed monkey genes (Rrox_v1)","Shrew genes (sorAra1)","Gilthead seabream genes (fSpaAur1.1)","Bolivian squirrel monkey genes (SaiBol1.0)","Common canary genes (SCA1)","African ostrich genes (ASM69896v1)","Saccharomyces cerevisiae genes (R64-1-1)","Greater amberjack genes (Sdu_1.0)","Asian bonytongue genes (fSclFor1.1)","Kakapo genes (bStrHab1_v1.p)","Tasmanian devil genes (mSarHar1.11)","Yellowtail amberjack genes (Sedor1)","Pike-perch genes (SLUC_FBN_1)","Turbot genes (ASM1334776v1)","Argentine black and white tegu genes (HLtupMer3)","Bicolor damselfish genes (Stegastes_partitus-1.0.2)","Tuatara genes (ASM311381v1)","Atlantic salmon genes (Ssal_v3.1)","Pig - Bamei genes (Bamei_pig_v1)","Pig - Berkshire genes (Berkshire_pig_v1)","Pig genes (Sscrofa11.1)","Pig - Hampshire genes (Hampshire_pig_v1)","Pig - Jinhua genes (Jinhua_pig_v1)","Pig - Landrace genes (Landrace_pig_v1)","Pig - Largewhite genes (Large_White_v1)","Pig - Meishan genes (Meishan_pig_v1)","Pig - Pietrain genes (Pietrain_pig_v1)","Pig - Rongchang genes (Rongchang_pig_v1)","Pig - Tibetan genes (Tibetan_Pig_v2)","Pig USMARC genes (USMARCv1.0)","Pig - Wuzhishan genes (minipig_v1.0)","Brown trout genes (fSalTru1.1)","Eurasian red squirrel genes (mSciVul1.1)","Tree Shrew genes (tupBel1)","Three-toed box turtle genes (T_m_triunguis-2.0)","Zebra finch genes (bTaeGut1_v1.p)","Tetraodon genes (TETRAODON 8.0)","Fugu genes (fTakRub1.2)","Dolphin genes (turTru1)","American black bear genes (ASM334442v1)","Polar bear genes (UrsMar_1.0)","Arctic ground squirrel genes (ASM342692v1)","Alpaca genes (vicPac1)","Common wombat genes (bare-nosed_wombat_genome_assembly)","Red fox genes (VulVul2.2)","Monterrey platyfish genes (Xiphophorus_couchianus-4.0.1)","Platyfish genes (X_maculatus-5.0-male)","Tropical clawed frog genes (UCB_Xtro_10.0)"],["ASM259213v1","fAstCal1.2","AnoCar2.0v2","bAquChr1.2","Midas_v5","ASM200744v2","Astyanax_mexicanus-2.0","Anan_2.0","AmpOce1.0","Nemo_v1","ASM874695v1","ASM210954v1","CAU_duck1.0","fAnaTes1.2","Bison_UMD1.0","LU_Bosgru_v3.0","UOA_Brahman_1","mBalMus1.v2","BosGru_v2.0","fBetSpl5.2","ARS-UCD1.3","ASM359739v1","Caty_1.0","ASM336829v1","Cypcar_WagV4.0","CamDro2","WBcel235","CHOK1GS_HDv1","fCotGob3.1","Ch_v2.0.2","ARS1","choHof1","CEY_v1","Cebus_imitator-1.0","KH","mCalJac1.pat.X","Coturnix_japonica_2.0","ChiLan1.0","ASM325472v1","ROS_Cfam_1.0","fCycLum1.pri","Callorhinchus_milii-6.1.3","Chrysemys_picta_bellii-3.0.3","Cavpor3.0","CroPor_comp1","ChlSab1.1","CSAV 2.0","Cse_v1.0","Tarsius_syrichta-2.0.1","C_variegatus-1.0","CatWag_v2_BIUU_UCD","fDenClu1.1","dlabrax2021","ASM228892v3","BDGP6.46","Dasnov3.0","Dord_2.0","GRCz11","ASM1607732v2","Eburgeri_3.2","EquCab3.0","fErpCal1.1","Ee_SOAP_WITH_SSPACE","eriEur1","Eluc_v4","TENREC","FicAlb1.5","Felis_catus_9.0","Fundulus_heteroclitus-3.0.2","BROAD S1","rGopEvg1_v1.p","GeoFor_1.0","bGalGal1.mat.broiler.GRCg7b","gorGor4","gadMor3.0","AstBur1.0","H_comes_QL1_v1","Naked_mole-rat_maternal","ASM331708v1","GRCh38.p14","IpCoco_1.2","SpeTri2.0","JacJac1.0","ASM164957v1","Loxafr3.0","BallGen_V1","ASB_HGAPassembly_v1","LatCha1","L_crocea_2.0","latLat_1.0","ASM966780v1","LepOcu1","fMasArm1.2","MesAur1.0","CAROLI_EIJ_v1.1","ASM229v1","Macaca_fascicularis_6.0","Turkey_5.1","Mleu.le_1.0","Myoluc2.0","marMar2.1","NGI_Narwhal_1","MosMos_v2_BIUU_UCD","Mmul_10","fMyrMur1.1","Mmur_3.0","GRCm39","Mnem_1.0","MicOch1.0","PAHARI_EIJ_v1.1","MusPutFur1.0","MUSP714","SPRET_EiJ_v1","M_zebra_UMD2a","NeoBri1.0","Meug_1.0","Nfu_20140520","S.galili_v1.0","Nleu_3.0","Nana_v5","TS10Xv2-PRI","NNQGG.v01","mOrnAna1.p.v1","ARS-UI_Ramb_v2.0","Oar_v3.1","OryCun2.0","OctDeg1.0","OtoGar3","OJAV_1.1","Okis_V2","ASM223467v1","Om_v0.7.RACA","USDA_OmykA_1.1","O_niloticus_UMD_NMBU","OchPri2.0-Ens","ASM858656v1","Otsh_v1.0","Susie_PABv2","Panubis1.0","proCap1","ASM283717v2","phaCin_unsw_v4.1","Pcoq_1.0","Poecilia_formosa-5.1.2","PKINGS_0.1","P_latipinna-1.0","PanLeo1.0","Parus_major1.1","Pmarinus_7.0","HU_Pman_2.1","PodMur_1.0","Pygocentrus_nattereri-1.0.2","PunNye1.0","panpan1.1","PanPar1.0","Guppy_female_1.0_MT","Prosim_1.0","PelSin_1.0","mPhoSin1.pri","PanTig1.0","EBS10Xv2-PRI","Pan_tro_3.0","pteVam1","ASM169854v1","mRhiFer1_v1.p","mRatBN7.2","Rrox_v1","sorAra1","fSpaAur1.1","SaiBol1.0","SCA1","ASM69896v1","R64-1-1","Sdu_1.0","fSclFor1.1","bStrHab1_v1.p","mSarHar1.11","Sedor1","SLUC_FBN_1","ASM1334776v1","HLtupMer3","Stegastes_partitus-1.0.2","ASM311381v1","Ssal_v3.1","Bamei_pig_v1","Berkshire_pig_v1","Sscrofa11.1","Hampshire_pig_v1","Jinhua_pig_v1","Landrace_pig_v1","Large_White_v1","Meishan_pig_v1","Pietrain_pig_v1","Rongchang_pig_v1","Tibetan_Pig_v2","USMARCv1.0","minipig_v1.0","fSalTru1.1","mSciVul1.1","tupBel1","T_m_triunguis-2.0","bTaeGut1_v1.p","TETRAODON 8.0","fTakRub1.2","turTru1","ASM334442v1","UrsMar_1.0","ASM342692v1","vicPac1","bare-nosed_wombat_genome_assembly","VulVul2.2","Xiphophorus_couchianus-4.0.1","X_maculatus-5.0-male","UCB_Xtro_10.0"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>dataset<\/th>\n      <th>description<\/th>\n      <th>version<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<p>Since we want to convert gene symbol to Entrez gene IDs in human subjects,
we select <code>biomart = 'genes'</code> and <code>dataset= 'hsapiens_gene_ensembl'</code>.
Use <code>useEnsembl</code> function to setup the connection with ensembl site host.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="query-bioloical-pathway-databases.html#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># convert from gene symbol to gene ID</span></span>
<span id="cb23-2"><a href="query-bioloical-pathway-databases.html#cb23-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-3"><a href="query-bioloical-pathway-databases.html#cb23-3" aria-hidden="true" tabindex="-1"></a>ensembl <span class="ot">&lt;-</span> <span class="fu">useEnsembl</span>(</span>
<span id="cb23-4"><a href="query-bioloical-pathway-databases.html#cb23-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">biomart =</span> <span class="st">&quot;genes&quot;</span>,</span>
<span id="cb23-5"><a href="query-bioloical-pathway-databases.html#cb23-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">dataset =</span> <span class="st">&quot;hsapiens_gene_ensembl&quot;</span>,</span>
<span id="cb23-6"><a href="query-bioloical-pathway-databases.html#cb23-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">mirror =</span> <span class="st">&quot;www&quot;</span></span>
<span id="cb23-7"><a href="query-bioloical-pathway-databases.html#cb23-7" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
</div>
<div id="annotate-genes" class="section level3 hasAnchor" number="4.2.2">
<h3><span class="header-section-number">4.2.2</span> Annotate genes<a href="query-bioloical-pathway-databases.html#annotate-genes" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Here instead of selecting genes in a biological pathway, I simply randomly
choose some genes in the dataset.</p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="query-bioloical-pathway-databases.html#cb24-1" aria-hidden="true" tabindex="-1"></a><span class="fu">set.seed</span>(<span class="dv">123</span>)</span>
<span id="cb24-2"><a href="query-bioloical-pathway-databases.html#cb24-2" aria-hidden="true" tabindex="-1"></a>gs1 <span class="ot">&lt;-</span> <span class="fu">sample</span>(rst<span class="sc">$</span>ID, <span class="dv">10</span>)</span>
<span id="cb24-3"><a href="query-bioloical-pathway-databases.html#cb24-3" aria-hidden="true" tabindex="-1"></a>gs2 <span class="ot">&lt;-</span> <span class="fu">sample</span>(rst<span class="sc">$</span>ID, <span class="dv">10</span>)</span></code></pre></div>
<p>Here I choose five most significantly differentially expressed genes,
five genes not differentially expressed.</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="query-bioloical-pathway-databases.html#cb25-1" aria-hidden="true" tabindex="-1"></a>gs3 <span class="ot">&lt;-</span> <span class="fu">c</span>(<span class="fu">head</span>(rst<span class="sc">$</span>ID, <span class="dv">5</span>), <span class="fu">tail</span>(rst<span class="sc">$</span>ID, <span class="dv">5</span>))</span>
<span id="cb25-2"><a href="query-bioloical-pathway-databases.html#cb25-2" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(glue<span class="sc">::</span><span class="fu">glue</span>(<span class="st">&#39;geneset 1: {paste0(gs1, collapse = &quot;, &quot;)}</span></span>
<span id="cb25-3"><a href="query-bioloical-pathway-databases.html#cb25-3" aria-hidden="true" tabindex="-1"></a><span class="st">geneset 2: {paste0(gs2, collapse =&quot;, &quot;)}</span><span class="sc">\n</span><span class="st">geneset 3: {paste0(gs3, collapse =&quot;, &quot;)}&#39;</span>))</span></code></pre></div>
<pre><code>geneset 1: HAUS5, MYDGF, TRAPPC13, LINS1, SLC14A1, DYRK3, EIF3I, NRP2, 207479_at, SESN1
geneset 2: ALDH6A1, MAPKAPK5-AS1, WNT7A, GH1, SMYD2, NT5DC2, CCDC30, MTSS1L, CETP, DZIP3
geneset 3: EBP, H3F3AP4, RAB2A, LOC100506248, MFSD11, NAP1L3, C19orf54, ADGRG6, PHF1, BCL2L14</code></pre>
<p>Retreive the annotated file with <code>getBM</code> function. Since we use gene symbols
to match aginst gene IDs, we’ll need to specify the type of input data in
the <code>filter</code> argument (<code>filters = "hgnc_symbol"</code>). We would like to output
the <code>hgnc_symbol</code>,<code>entrezgene_id</code> reference.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="query-bioloical-pathway-databases.html#cb27-1" aria-hidden="true" tabindex="-1"></a>annotgs3 <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb27-2"><a href="query-bioloical-pathway-databases.html#cb27-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">&quot;hgnc_symbol&quot;</span>, <span class="st">&quot;entrezgene_id&quot;</span>),</span>
<span id="cb27-3"><a href="query-bioloical-pathway-databases.html#cb27-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">filters =</span> <span class="st">&quot;hgnc_symbol&quot;</span>,</span>
<span id="cb27-4"><a href="query-bioloical-pathway-databases.html#cb27-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">values =</span> gs3,</span>
<span id="cb27-5"><a href="query-bioloical-pathway-databases.html#cb27-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">mart =</span> ensembl</span>
<span id="cb27-6"><a href="query-bioloical-pathway-databases.html#cb27-6" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb27-7"><a href="query-bioloical-pathway-databases.html#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(annotgs3)</span></code></pre></div>
<pre><code>  hgnc_symbol entrezgene_id
1      ADGRG6         57211
2     BCL2L14         79370
3         EBP         10682
4      MFSD11         79157
5      NAP1L3          4675
6        PHF1          5252
7       RAB2A          5862</code></pre>
<p><strong>Note that not all gene symbols can be perfectly match against gene ID.</strong>
Alternatively, you can annotate all the gene in the dataset then extract
the ones in the genesets. For simplicity, I removed duplicated genes of which
have multiple probes correspond to the same gene (isoforms). In practice, the
removal of these isoforms should be done with examination of coefficient
variation (CV).</p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="query-bioloical-pathway-databases.html#cb29-1" aria-hidden="true" tabindex="-1"></a>all_genes <span class="ot">&lt;-</span> rst<span class="sc">$</span>ID[<span class="sc">!</span><span class="fu">duplicated</span>(rst<span class="sc">$</span>ID)]</span></code></pre></div>
<p>Upon removal of duplicates, there are 14063 genes left in
the dataset. We annotate these genes.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="query-bioloical-pathway-databases.html#cb30-1" aria-hidden="true" tabindex="-1"></a>annot_all <span class="ot">&lt;-</span> <span class="fu">getBM</span>(</span>
<span id="cb30-2"><a href="query-bioloical-pathway-databases.html#cb30-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">attributes =</span> <span class="fu">c</span>(<span class="st">&quot;hgnc_symbol&quot;</span>, <span class="st">&quot;entrezgene_id&quot;</span>),</span>
<span id="cb30-3"><a href="query-bioloical-pathway-databases.html#cb30-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">filters =</span> <span class="st">&quot;hgnc_symbol&quot;</span>,</span>
<span id="cb30-4"><a href="query-bioloical-pathway-databases.html#cb30-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">values =</span> all_genes,</span>
<span id="cb30-5"><a href="query-bioloical-pathway-databases.html#cb30-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">mart =</span> ensembl</span>
<span id="cb30-6"><a href="query-bioloical-pathway-databases.html#cb30-6" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
</div>
<div id="alternative-methods-for-annotation" class="section level3 hasAnchor" number="4.2.3">
<h3><span class="header-section-number">4.2.3</span> Alternative methods for annotation<a href="query-bioloical-pathway-databases.html#alternative-methods-for-annotation" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>Alternatively one can use <code>bitr</code> function from <code>clusterProfiler</code> package.
Here we need to specify the database for curation. This requires the pacakge
<code>org.Hs.eg.db</code>. If you decide to go with method, make sure to update this
package often to obtain the most current information. The <code>org.Hs.eg.db</code>
package is updated biannually.</p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="query-bioloical-pathway-databases.html#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="fu">bitr</span>(gs1,</span>
<span id="cb31-2"><a href="query-bioloical-pathway-databases.html#cb31-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">fromType =</span> <span class="st">&quot;SYMBOL&quot;</span>,</span>
<span id="cb31-3"><a href="query-bioloical-pathway-databases.html#cb31-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">toType =</span> <span class="fu">c</span>(<span class="st">&quot;ENTREZID&quot;</span>, <span class="st">&quot;SYMBOL&quot;</span>),</span>
<span id="cb31-4"><a href="query-bioloical-pathway-databases.html#cb31-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">OrgDb =</span> org.Hs.eg.db</span>
<span id="cb31-5"><a href="query-bioloical-pathway-databases.html#cb31-5" aria-hidden="true" tabindex="-1"></a>)</span></code></pre></div>
<pre><code>     SYMBOL ENTREZID
1     HAUS5    23354
2     MYDGF    56005
3  TRAPPC13    80006
4     LINS1    55180
5   SLC14A1     6563
6     DYRK3     8444
7     EIF3I     8668
8      NRP2     8828
10    SESN1    27244</code></pre>
</div>
</div>
<div id="curate-biological-functions" class="section level2 hasAnchor" number="4.3">
<h2><span class="header-section-number">4.3</span> Curate biological functions<a href="query-bioloical-pathway-databases.html#curate-biological-functions" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Upon selection genes of interest, we can start to curate the biological
functions of these genes. The most popular databases are GO, KEGG, Reactome,
etc. For full understanding the property of each of these databases, see
these references [xxx].</p>
<p>For example, I’d like to query the biological functions of genes in the set 3
. One can use the <code>groupGO</code> function to look up biological functions:</p>
<blockquote>
<ul>
<li>MF: Molecular Function
<ul>
<li>molecular activities of gene products</li>
</ul></li>
<li>CC: Cellular Component
<ul>
<li>where gene products are active</li>
</ul></li>
<li>BP: Biological Process
<ul>
<li>pathways and larger processes made up of the activities of multiple
gene products</li>
</ul></li>
</ul>
</blockquote>
<p>Additionally, one can look up onto <a href="https://www.genome.jp/kegg/ko.html">KEGG</a>,
<a href="https://www.wikipathways.org/">WikiPathway</a>, <a href="https://reactome.org/">Reactome</a>,
<a href="https://disease-ontology.org/">Diease Ontology (DO)</a>,
<a href="https://www.nlm.nih.gov/mesh/meshhome.html">Medical Subject Headings (MeSH)</a>,
<a href="https://www.gsea-msigdb.org/gsea/msigdb">Molecular Signatures Database (MSigDb)</a>.
I didn’t find any ready to use functions from <code>clusterProfile</code> package,except
for <code>GO</code> query. However, personally, I don’t think statiticians are the experts to
interpret the finding in biological sense. Since this is less of interest for
analysts, the <code>clusterProfiler</code> simplifed this curation process and combined
with the tests (ORA, GSEA) in their function construction.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="query-bioloical-pathway-databases.html#cb33-1" aria-hidden="true" tabindex="-1"></a>ggo <span class="ot">&lt;-</span> <span class="fu">groupGO</span>(</span>
<span id="cb33-2"><a href="query-bioloical-pathway-databases.html#cb33-2" aria-hidden="true" tabindex="-1"></a>  <span class="at">gene =</span> <span class="fu">as.character</span>(annotgs3<span class="sc">$</span>entrezgene_id),</span>
<span id="cb33-3"><a href="query-bioloical-pathway-databases.html#cb33-3" aria-hidden="true" tabindex="-1"></a>  <span class="at">OrgDb =</span> org.Hs.eg.db,</span>
<span id="cb33-4"><a href="query-bioloical-pathway-databases.html#cb33-4" aria-hidden="true" tabindex="-1"></a>  <span class="at">ont =</span> <span class="st">&quot;BP&quot;</span>,</span>
<span id="cb33-5"><a href="query-bioloical-pathway-databases.html#cb33-5" aria-hidden="true" tabindex="-1"></a>  <span class="at">level =</span> <span class="dv">3</span>,</span>
<span id="cb33-6"><a href="query-bioloical-pathway-databases.html#cb33-6" aria-hidden="true" tabindex="-1"></a>  <span class="at">readable =</span> <span class="cn">TRUE</span></span>
<span id="cb33-7"><a href="query-bioloical-pathway-databases.html#cb33-7" aria-hidden="true" tabindex="-1"></a>)</span>
<span id="cb33-8"><a href="query-bioloical-pathway-databases.html#cb33-8" aria-hidden="true" tabindex="-1"></a><span class="fu">tail</span>(ggo)</span></code></pre></div>
<pre><code>                   ID
GO:0051410 GO:0051410
GO:0061687 GO:0061687
GO:0071722 GO:0071722
GO:0140330 GO:0140330
GO:1990961 GO:1990961
GO:0031214 GO:0031214
                                                                                Description Count
GO:0051410                                              detoxification of nitrogen compound     0
GO:0061687                                             detoxification of inorganic compound     0
GO:0071722                                   detoxification of arsenic-containing substance     0
GO:0140330 xenobiotic detoxification by transmembrane export across the cell outer membrane     0
GO:1990961     xenobiotic detoxification by transmembrane export across the plasma membrane     0
GO:0031214                                                    biomineral tissue development     0
           GeneRatio geneID
GO:0051410       0/8       
GO:0061687       0/8       
GO:0071722       0/8       
GO:0140330       0/8       
GO:1990961       0/8       
GO:0031214       0/8       </code></pre>
</div>
<div id="reference" class="section level2 hasAnchor" number="4.4">
<h2><span class="header-section-number">4.4</span> Reference<a href="query-bioloical-pathway-databases.html#reference" class="anchor-section" aria-label="Anchor link to header"></a></h2>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="differential-gene-expression-analysis-dgea.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="over-representation-analysis.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
