<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title>2 Study dataset | Pathway analysis methods</title>
  <meta name="description" content="This notebook is dedicated to genetic pathway analysis methods." />
  <meta name="generator" content="bookdown 0.34 and GitBook 2.6.7" />

  <meta property="og:title" content="2 Study dataset | Pathway analysis methods" />
  <meta property="og:type" content="book" />
  
  <meta property="og:description" content="This notebook is dedicated to genetic pathway analysis methods." />
  

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="2 Study dataset | Pathway analysis methods" />
  
  <meta name="twitter:description" content="This notebook is dedicated to genetic pathway analysis methods." />
  

<meta name="author" content="Siying Huang (E0482362)" />


<meta name="date" content="2024-01-19" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="gene-sets-and-pathways.html"/>
<link rel="next" href="differential-gene-expression-analysis-dgea.html"/>
<script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<script src="https://cdn.jsdelivr.net/npm/fuse.js@6.4.6/dist/fuse.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />








<link href="libs/anchor-sections-1.1.0/anchor-sections.css" rel="stylesheet" />
<link href="libs/anchor-sections-1.1.0/anchor-sections-hash.css" rel="stylesheet" />
<script src="libs/anchor-sections-1.1.0/anchor-sections.js"></script>
<script src="libs/htmlwidgets-1.6.2/htmlwidgets.js"></script>
<link href="libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="libs/datatables-binding-0.26/datatables.js"></script>
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="libs/dt-core-1.12.1/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="libs/dt-core-1.12.1/js/jquery.dataTables.min.js"></script>
<link href="libs/crosstalk-1.1.1/css/crosstalk.css" rel="stylesheet" />
<script src="libs/crosstalk-1.1.1/js/crosstalk.min.js"></script>


<style type="text/css">
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { display: inline-block; line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

<style type="text/css">
  
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="index.html"><a href="index.html"><i class="fa fa-check"></i>Prefeace</a></li>
<li class="chapter" data-level="1" data-path="gene-sets-and-pathways.html"><a href="gene-sets-and-pathways.html"><i class="fa fa-check"></i><b>1</b> Gene sets and pathways</a>
<ul>
<li class="chapter" data-level="1.1" data-path="gene-sets-and-pathways.html"><a href="gene-sets-and-pathways.html#organization-of-this-e-book"><i class="fa fa-check"></i><b>1.1</b> Organization of this e-book</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="study-dataset.html"><a href="study-dataset.html"><i class="fa fa-check"></i><b>2</b> Study dataset</a>
<ul>
<li class="chapter" data-level="2.1" data-path="study-dataset.html"><a href="study-dataset.html#extracting-data-from-ncbi-gene-expression-omnibus-databse"><i class="fa fa-check"></i><b>2.1</b> Extracting data from NCBI Gene Expression Omnibus databse</a>
<ul>
<li class="chapter" data-level="2.1.1" data-path="study-dataset.html"><a href="study-dataset.html#summary-of-the-study"><i class="fa fa-check"></i><b>2.1.1</b> Summary of the study</a></li>
<li class="chapter" data-level="2.1.2" data-path="study-dataset.html"><a href="study-dataset.html#assigning-the-expression-and-phenotype-data"><i class="fa fa-check"></i><b>2.1.2</b> Assigning the expression and phenotype data</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="study-dataset.html"><a href="study-dataset.html#convert-to-gds-to-expression-set"><i class="fa fa-check"></i><b>2.2</b> convert to GDS to expression set</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="differential-gene-expression-analysis-dgea.html"><a href="differential-gene-expression-analysis-dgea.html"><i class="fa fa-check"></i><b>3</b> Differential gene expression analysis (DGEA)</a>
<ul>
<li class="chapter" data-level="3.1" data-path="differential-gene-expression-analysis-dgea.html"><a href="differential-gene-expression-analysis-dgea.html#comparison-between-smokers-and-non-smokers"><i class="fa fa-check"></i><b>3.1</b> Comparison between smokers and non-smokers</a></li>
<li class="chapter" data-level="3.2" data-path="differential-gene-expression-analysis-dgea.html"><a href="differential-gene-expression-analysis-dgea.html#results-visualization"><i class="fa fa-check"></i><b>3.2</b> Results visualization</a></li>
</ul></li>
<li class="chapter" data-level="4" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html"><i class="fa fa-check"></i><b>4</b> Query bioloical pathway databases</a>
<ul>
<li class="chapter" data-level="4.1" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#gene-annotation"><i class="fa fa-check"></i><b>4.1</b> Gene annotation</a></li>
<li class="chapter" data-level="4.2" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#convert-gene-symbol-to-gene-id-using-biomart."><i class="fa fa-check"></i><b>4.2</b> Convert gene symbol to gene ID using <code>biomaRt</code>.</a>
<ul>
<li class="chapter" data-level="4.2.1" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#connect-to-the-selected-biomart-database-using-useensembl-function."><i class="fa fa-check"></i><b>4.2.1</b> Connect to the selected <em>BioMart</em> database using <code>useEnsembl</code> function.</a></li>
<li class="chapter" data-level="4.2.2" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#annotate-genes"><i class="fa fa-check"></i><b>4.2.2</b> Annotate genes</a></li>
<li class="chapter" data-level="4.2.3" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#alternative-methods-for-annotation"><i class="fa fa-check"></i><b>4.2.3</b> Alternative methods for annotation</a></li>
</ul></li>
<li class="chapter" data-level="4.3" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#curate-biological-functions"><i class="fa fa-check"></i><b>4.3</b> Curate biological functions</a></li>
<li class="chapter" data-level="4.4" data-path="query-bioloical-pathway-databases.html"><a href="query-bioloical-pathway-databases.html#reference"><i class="fa fa-check"></i><b>4.4</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="5" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html"><i class="fa fa-check"></i><b>5</b> Over-representation Analysis</a>
<ul>
<li class="chapter" data-level="5.1" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#data-prepartion"><i class="fa fa-check"></i><b>5.1</b> Data prepartion</a></li>
<li class="chapter" data-level="5.2" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#gene-set-selection"><i class="fa fa-check"></i><b>5.2</b> Gene set selection</a>
<ul>
<li class="chapter" data-level="5.2.1" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#define-differentially-expressed-gene"><i class="fa fa-check"></i><b>5.2.1</b> Define differentially expressed gene</a></li>
</ul></li>
<li class="chapter" data-level="5.3" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#over-representation-analysis-1"><i class="fa fa-check"></i><b>5.3</b> Over-representation analysis</a></li>
<li class="chapter" data-level="5.4" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#over-respensentation-analysis-from-r-package"><i class="fa fa-check"></i><b>5.4</b> Over-respensentation analysis from R package</a></li>
<li class="chapter" data-level="5.5" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#test-set-up-and-result-interpretations"><i class="fa fa-check"></i><b>5.5</b> Test set up and result interpretations</a></li>
<li class="chapter" data-level="5.6" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#examples"><i class="fa fa-check"></i><b>5.6</b> Examples</a>
<ul>
<li class="chapter" data-level="5.6.1" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#go"><i class="fa fa-check"></i><b>5.6.1</b> GO</a></li>
<li class="chapter" data-level="5.6.2" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#kegg"><i class="fa fa-check"></i><b>5.6.2</b> KEGG</a></li>
<li class="chapter" data-level="5.6.3" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#wikipathways"><i class="fa fa-check"></i><b>5.6.3</b> WikiPathways</a></li>
<li class="chapter" data-level="5.6.4" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#reactome"><i class="fa fa-check"></i><b>5.6.4</b> Reactome</a></li>
<li class="chapter" data-level="5.6.5" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#do"><i class="fa fa-check"></i><b>5.6.5</b> DO</a></li>
<li class="chapter" data-level="5.6.6" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#disgenet"><i class="fa fa-check"></i><b>5.6.6</b> DisGeNET</a></li>
<li class="chapter" data-level="5.6.7" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#mesh"><i class="fa fa-check"></i><b>5.6.7</b> MeSH</a></li>
</ul></li>
<li class="chapter" data-level="5.7" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#pros-and-cons-of-ora"><i class="fa fa-check"></i><b>5.7</b> Pros and Cons of ORA</a></li>
<li class="chapter" data-level="5.8" data-path="over-representation-analysis.html"><a href="over-representation-analysis.html#reference-1"><i class="fa fa-check"></i><b>5.8</b> Reference</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="appendix.html"><a href="appendix.html"><i class="fa fa-check"></i>Appendix</a>
<ul>
<li class="chapter" data-level="5.9" data-path="appendix.html"><a href="appendix.html#biological-process-data-bases"><i class="fa fa-check"></i><b>5.9</b> Biological process data bases</a></li>
<li class="chapter" data-level="5.10" data-path="appendix.html"><a href="appendix.html#summary-of-currently-available-pathway-analysis-methods"><i class="fa fa-check"></i><b>5.10</b> Summary of currently available pathway analysis methods</a>
<ul>
<li class="chapter" data-level="5.10.1" data-path="appendix.html"><a href="appendix.html#over-representation-analysis-2"><i class="fa fa-check"></i><b>5.10.1</b> Over-representation analysis</a></li>
<li class="chapter" data-level="5.10.2" data-path="appendix.html"><a href="appendix.html#gene-set-enrichment-analysis"><i class="fa fa-check"></i><b>5.10.2</b> Gene set enrichment analysis</a></li>
</ul></li>
</ul></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Pathway analysis methods</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="study-dataset" class="section level1 hasAnchor" number="2">
<h1><span class="header-section-number">2</span> Study dataset<a href="study-dataset.html#study-dataset" class="anchor-section" aria-label="Anchor link to header"></a></h1>
<p>In this tutorial, I still use the pulic smoking study dataset from NCBI
Gene Expression Omnibus (GEO) <a href="https://www.ncbi.nlm.nih.gov/gds">database</a> - GDS3713 (GSE18723).</p>
<div id="extracting-data-from-ncbi-gene-expression-omnibus-databse" class="section level2 hasAnchor" number="2.1">
<h2><span class="header-section-number">2.1</span> Extracting data from NCBI Gene Expression Omnibus databse<a href="study-dataset.html#extracting-data-from-ncbi-gene-expression-omnibus-databse" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Using <code>GEOquery</code> library to retrieve data from NCBI database. The GDS dataset
is stored in the <code>gds</code> R4 class object.</p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="study-dataset.html#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb2-2"><a href="study-dataset.html#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(GEOquery)</span>
<span id="cb2-3"><a href="study-dataset.html#cb2-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb2-4"><a href="study-dataset.html#cb2-4" aria-hidden="true" tabindex="-1"></a>gds <span class="ot">&lt;-</span> <span class="fu">getGEO</span>(<span class="st">&quot;GDS3713&quot;</span>)</span>
<span id="cb2-5"><a href="study-dataset.html#cb2-5" aria-hidden="true" tabindex="-1"></a><span class="co"># gds@header # meta data of the experiment</span></span>
<span id="cb2-6"><a href="study-dataset.html#cb2-6" aria-hidden="true" tabindex="-1"></a><span class="co"># gds@dataTable@table # expression data</span></span>
<span id="cb2-7"><a href="study-dataset.html#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co"># gds@dataTable@columns # phenotype data</span></span></code></pre></div>
<p>structure of the <code>gds</code> object</p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="study-dataset.html#cb3-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(gds, <span class="at">max.level =</span> <span class="dv">3</span>)</span></code></pre></div>
<pre><code>Formal class &#39;GDS&#39; [package &quot;GEOquery&quot;] with 3 slots
  ..@ gpl      :Formal class &#39;GPL&#39; [package &quot;GEOquery&quot;] with 2 slots
  ..@ dataTable:Formal class &#39;GEODataTable&#39; [package &quot;GEOquery&quot;] with 2 slots
  ..@ header   :List of 23
  .. ..$ channel_count           : chr &quot;1&quot;
  .. ..$ dataset_id              : chr [1:2] &quot;GDS3713&quot; &quot;GDS3713&quot;
  .. ..$ description             : chr [1:3] &quot;Analysis of peripheral circulating B cells from smoking and non-smoking healthy US white females. B cells are d&quot;| __truncated__ &quot;control&quot; &quot;cigarette smoke&quot;
  .. ..$ email                   : chr &quot;geo@ncbi.nlm.nih.gov&quot;
  .. ..$ feature_count           : chr &quot;22283&quot;
  .. ..$ institute               : chr &quot;NCBI NLM NIH&quot;
  .. ..$ name                    : chr &quot;Gene Expression Omnibus (GEO)&quot;
  .. ..$ order                   : chr &quot;none&quot;
  .. ..$ platform                : chr &quot;GPL96&quot;
  .. ..$ platform_organism       : chr &quot;Homo sapiens&quot;
  .. ..$ platform_technology_type: chr &quot;in situ oligonucleotide&quot;
  .. ..$ pubmed_id               : chr &quot;20217071&quot;
  .. ..$ ref                     : chr &quot;Nucleic Acids Res. 2005 Jan 1;33 Database Issue:D562-6&quot;
  .. ..$ reference_series        : chr &quot;GSE18723&quot;
  .. ..$ sample_count            : chr &quot;79&quot;
  .. ..$ sample_id               : chr [1:2] &quot;GSM464697,GSM464698,GSM464699,GSM464700,GSM464701,GSM464702,GSM464703,GSM464704,GSM464705,GSM464706,GSM464707,G&quot;| __truncated__ &quot;GSM464658,GSM464659,GSM464660,GSM464661,GSM464662,GSM464663,GSM464664,GSM464665,GSM464666,GSM464667,GSM464668,G&quot;| __truncated__
  .. ..$ sample_organism         : chr &quot;Homo sapiens&quot;
  .. ..$ sample_type             : chr &quot;RNA&quot;
  .. ..$ title                   : chr &quot;Smoking effect on B lymphocytes in females&quot;
  .. ..$ type                    : chr [1:3] &quot;Expression profiling by array&quot; &quot;stress&quot; &quot;stress&quot;
  .. ..$ update_date             : chr &quot;Jul 11 2010&quot;
  .. ..$ value_type              : chr &quot;count&quot;
  .. ..$ web_link                : chr &quot;http://www.ncbi.nlm.nih.gov/geo&quot;</code></pre>
<div id="summary-of-the-study" class="section level3 hasAnchor" number="2.1.1">
<h3><span class="header-section-number">2.1.1</span> Summary of the study<a href="study-dataset.html#summary-of-the-study" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>This study aim to analyze peripheral circulating B cells from smoking and non-smoking healthy US white females. B cells are directly associated with the onset and development of many smoking-induced diseases. Results provide insight into the molecular basis of B cell involvement in smoking-related pathogenesis. It contains 39 smoker and 40 non-smokers.</p>
</div>
<div id="assigning-the-expression-and-phenotype-data" class="section level3 hasAnchor" number="2.1.2">
<h3><span class="header-section-number">2.1.2</span> Assigning the expression and phenotype data<a href="study-dataset.html#assigning-the-expression-and-phenotype-data" class="anchor-section" aria-label="Anchor link to header"></a></h3>
<p>The expression data is stored in the <code>gds@dataTable@tabl</code> with gene in
The rows and subjects in the columns.
The phenotype data is store at <code>gds@dataTable@columns</code>.</p>
<div class="sourceCode" id="cb5"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb5-1"><a href="study-dataset.html#cb5-1" aria-hidden="true" tabindex="-1"></a>expr <span class="ot">&lt;-</span> <span class="fu">Table</span>(gds)</span>
<span id="cb5-2"><a href="study-dataset.html#cb5-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(expr[, <span class="dv">1</span><span class="sc">:</span><span class="dv">4</span>])</span></code></pre></div>
<pre><code>    ID_REF           IDENTIFIER          GSM464697          GSM464698       
 Length:22283       Length:22283       Min.   :    1.53   Min.   :    1.06  
 Class :character   Class :character   1st Qu.:   65.44   1st Qu.:   74.40  
 Mode  :character   Mode  :character   Median :  221.78   Median :  243.15  
                                       Mean   :  859.82   Mean   :  752.89  
                                       3rd Qu.:  582.95   3rd Qu.:  600.85  
                                       Max.   :51356.90   Max.   :34387.10  </code></pre>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="study-dataset.html#cb7-1" aria-hidden="true" tabindex="-1"></a>pheno <span class="ot">&lt;-</span> <span class="fu">Columns</span>(gds)</span>
<span id="cb7-2"><a href="study-dataset.html#cb7-2" aria-hidden="true" tabindex="-1"></a><span class="fu">summary</span>(pheno)</span></code></pre></div>
<pre><code>    sample                      stress   description       
 Length:79          cigarette smoke:39   Length:79         
 Class :character   control        :40   Class :character  
 Mode  :character                        Mode  :character  </code></pre>
</div>
</div>
<div id="convert-to-gds-to-expression-set" class="section level2 hasAnchor" number="2.2">
<h2><span class="header-section-number">2.2</span> convert to GDS to expression set<a href="study-dataset.html#convert-to-gds-to-expression-set" class="anchor-section" aria-label="Anchor link to header"></a></h2>
<p>Converting the gds object to expression set will generate a series of sub documents, including an annotation file of the genes in the expression data.</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="study-dataset.html#cb9-1" aria-hidden="true" tabindex="-1"></a>eset <span class="ot">&lt;-</span> <span class="fu">GDS2eSet</span>(gds)</span>
<span id="cb9-2"><a href="study-dataset.html#cb9-2" aria-hidden="true" tabindex="-1"></a><span class="fu">dim</span>(<span class="fu">pData</span>(eset)) <span class="co"># pheno data, data.frame</span></span></code></pre></div>
<pre><code>[1] 79  3</code></pre>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="study-dataset.html#cb11-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(eset, <span class="at">max.level =</span><span class="dv">4</span>)</span></code></pre></div>
<pre><code>Formal class &#39;ExpressionSet&#39; [package &quot;Biobase&quot;] with 7 slots
  ..@ experimentData   :Formal class &#39;MIAME&#39; [package &quot;Biobase&quot;] with 13 slots
  .. .. ..@ name             : chr &quot;&quot;
  .. .. ..@ lab              : chr &quot;&quot;
  .. .. ..@ contact          : chr &quot;&quot;
  .. .. ..@ title            : chr &quot;Smoking effect on B lymphocytes in females&quot;
  .. .. ..@ abstract         : chr &quot;Analysis of peripheral circulating B cells from smoking and non-smoking healthy US white females. B cells are d&quot;| __truncated__
  .. .. ..@ url              : chr &quot;&quot;
  .. .. ..@ pubMedIds        : chr &quot;20217071&quot;
  .. .. ..@ samples          : list()
  .. .. ..@ hybridizations   : list()
  .. .. ..@ normControls     : list()
  .. .. ..@ preprocessing    : list()
  .. .. ..@ other            :List of 23
  .. .. ..@ .__classVersion__:Formal class &#39;Versions&#39; [package &quot;Biobase&quot;] with 1 slot
  ..@ assayData        :&lt;environment: 0x2d480628&gt; 
  ..@ phenoData        :Formal class &#39;AnnotatedDataFrame&#39; [package &quot;Biobase&quot;] with 4 slots
  .. .. ..@ varMetadata      :&#39;data.frame&#39;: 3 obs. of  1 variable:
  .. .. ..@ data             :&#39;data.frame&#39;: 79 obs. of  3 variables:
  .. .. ..@ dimLabels        : chr [1:2] &quot;sampleNames&quot; &quot;sampleColumns&quot;
  .. .. ..@ .__classVersion__:Formal class &#39;Versions&#39; [package &quot;Biobase&quot;] with 1 slot
  ..@ featureData      :Formal class &#39;AnnotatedDataFrame&#39; [package &quot;Biobase&quot;] with 4 slots
  .. .. ..@ varMetadata      :&#39;data.frame&#39;: 21 obs. of  2 variables:
  .. .. ..@ data             :&#39;data.frame&#39;: 22283 obs. of  21 variables:
  .. .. ..@ dimLabels        : chr [1:2] &quot;featureNames&quot; &quot;featureColumns&quot;
  .. .. ..@ .__classVersion__:Formal class &#39;Versions&#39; [package &quot;Biobase&quot;] with 1 slot
  ..@ annotation       : chr(0) 
  ..@ protocolData     :Formal class &#39;AnnotatedDataFrame&#39; [package &quot;Biobase&quot;] with 4 slots
  .. .. ..@ varMetadata      :&#39;data.frame&#39;: 0 obs. of  1 variable:
  .. .. ..@ data             :&#39;data.frame&#39;: 79 obs. of  0 variables
  .. .. ..@ dimLabels        : chr [1:2] &quot;sampleNames&quot; &quot;sampleColumns&quot;
  .. .. ..@ .__classVersion__:Formal class &#39;Versions&#39; [package &quot;Biobase&quot;] with 1 slot
  ..@ .__classVersion__:Formal class &#39;Versions&#39; [package &quot;Biobase&quot;] with 1 slot
  .. .. ..@ .Data:List of 4
  .. .. ..$ names: chr [1:4] &quot;R&quot; &quot;Biobase&quot; &quot;eSet&quot; &quot;ExpressionSet&quot;</code></pre>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="study-dataset.html#cb13-1" aria-hidden="true" tabindex="-1"></a><span class="co"># gene annotations</span></span>
<span id="cb13-2"><a href="study-dataset.html#cb13-2" aria-hidden="true" tabindex="-1"></a>annot_cols <span class="ot">&lt;-</span> eset<span class="sc">@</span>featureData<span class="sc">@</span>varMetadata</span>
<span id="cb13-3"><a href="study-dataset.html#cb13-3" aria-hidden="true" tabindex="-1"></a>annot_data <span class="ot">&lt;-</span> eset<span class="sc">@</span>featureData<span class="sc">@</span>data</span></code></pre></div>

</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="gene-sets-and-pathways.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="differential-gene-expression-analysis-dgea.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"whatsapp": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"search": {
"engine": "fuse",
"options": null
},
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
